// ng-uikit/__ivy_ngcc__/fesm2015/ng-uikit.js
import { Component, NgModule, ContentChildren, Input, EventEmitter, Output, ElementRef, Injectable, defineInjectable } from '@angular/core';
import { CommonModule } from '@angular/common';
import { HttpClient } from '@angular/common/http';
import { FormsModule } from '@angular/forms';
import { BrowserModule } from '@angular/platform-browser';

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
import * as ɵngcc0 from '@angular/core';
import * as ɵngcc1 from '@angular/common';
import * as ɵngcc2 from '@angular/common/http';
import * as ɵngcc3 from '@angular/forms';

const _c0 = ["*"];
function AccordionPanelComponent_span_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "span", 8);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r0 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r0.header, " ");
} }
function AccordionPanelComponent_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵprojection(0, 1, ["*ngIf", "hasHeader"]);
} }
const _c1 = ["*", [["ui-accordion-panel-header"]]];
const _c2 = function (a0) { return { "show": a0 }; };
const _c3 = ["*", "ui-accordion-panel-header"];
function AutocompleteComponent_input_3_Template(rf, ctx) { if (rf & 1) {
    const _r7 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 7);
    ɵngcc0.ɵɵlistener("keyup", function AutocompleteComponent_input_3_Template_input_keyup_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r7); const ctx_r6 = ɵngcc0.ɵɵnextContext(); return ctx_r6.filter($event.target.value); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r2 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate1("placeholder", "Enter ", ctx_r2.label, "");
    ɵngcc0.ɵɵproperty("ngModel", ctx_r2.selectedItem[ctx_r2.niddle]);
} }
function AutocompleteComponent_input_4_Template(rf, ctx) { if (rf & 1) {
    const _r9 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "input", 8);
    ɵngcc0.ɵɵlistener("keyup", function AutocompleteComponent_input_4_Template_input_keyup_0_listener($event) { ɵngcc0.ɵɵrestoreView(_r9); const ctx_r8 = ɵngcc0.ɵɵnextContext(); return ctx_r8.filter($event.target.value); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r3 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵpropertyInterpolate1("placeholder", "Enter ", ctx_r3.label, "");
    ɵngcc0.ɵɵproperty("ngModel", ctx_r3.selectedItem);
} }
function AutocompleteComponent_small_5_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "small", 9);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r4 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r4.helpText);
} }
function AutocompleteComponent_ul_7_li_1_Template(rf, ctx) { if (rf & 1) {
    const _r13 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 12);
    ɵngcc0.ɵɵlistener("click", function AutocompleteComponent_ul_7_li_1_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r13); const item_r11 = ctx.$implicit; const ctx_r12 = ɵngcc0.ɵɵnextContext(2); return ctx_r12.selectItem(item_r11); });
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r11 = ctx.$implicit;
    const ctx_r10 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(ctx_r10.getText(item_r11));
} }
function AutocompleteComponent_ul_7_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "ul", 10);
    ɵngcc0.ɵɵtemplate(1, AutocompleteComponent_ul_7_li_1_Template, 2, 1, "li", 11);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r5 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r5.filteredList);
} }
function CardComponent_div_0_img_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "img", 5);
} if (rf & 2) {
    const ctx_r15 = ɵngcc0.ɵɵnextContext(2);
    ɵngcc0.ɵɵpropertyInterpolate("src", ctx_r15.config.src, ɵngcc0.ɵɵsanitizeUrl);
    ɵngcc0.ɵɵpropertyInterpolate("alt", ctx_r15.config.alt || ctx_r15.config.title);
} }
function CardComponent_div_0_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 1);
    ɵngcc0.ɵɵtemplate(1, CardComponent_div_0_img_1_Template, 1, 2, "img", 2);
    ɵngcc0.ɵɵelementStart(2, "div", 3);
    ɵngcc0.ɵɵelementStart(3, "h5", 4);
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵprojection(5);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r14 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", ctx_r14.config.src);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(ctx_r14.config.title);
} }
const _c4 = function (a0) { return { "active": a0 }; };
function CarouselComponent_li_2_Template(rf, ctx) { if (rf & 1) {
    const _r21 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 10);
    ɵngcc0.ɵɵlistener("click", function CarouselComponent_li_2_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r21); const i_r19 = ctx.index; const ctx_r20 = ɵngcc0.ɵɵnextContext(); return ctx_r20.goTo(i_r19); });
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const i_r19 = ctx.index;
    const ctx_r16 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(2, _c4, i_r19 == ctx_r16.activeIndex));
    ɵngcc0.ɵɵattribute("data-slide-to", i_r19);
} }
function CarouselComponent_div_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 11);
    ɵngcc0.ɵɵelement(1, "img", 12);
    ɵngcc0.ɵɵelementStart(2, "div", 13);
    ɵngcc0.ɵɵelementStart(3, "h5");
    ɵngcc0.ɵɵtext(4);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "p");
    ɵngcc0.ɵɵtext(6);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r22 = ctx.$implicit;
    const i_r23 = ctx.index;
    const ctx_r17 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(5, _c4, i_r23 == ctx_r17.activeIndex));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("src", item_r22.src, ɵngcc0.ɵɵsanitizeUrl);
    ɵngcc0.ɵɵpropertyInterpolate("alt", item_r22.alt);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵtextInterpolate(item_r22.alt);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(item_r22.alt);
} }
const _c5 = function (a0) { return { width: a0 }; };
function DatatableComponent_div_2_Template(rf, ctx) { if (rf & 1) {
    const _r29 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementStart(2, "i", 7);
    ɵngcc0.ɵɵlistener("click", function DatatableComponent_div_2_Template_i_click_2_listener() { ɵngcc0.ɵɵrestoreView(_r29); const col_r27 = ctx.$implicit; const ctx_r28 = ɵngcc0.ɵɵnextContext(); return ctx_r28.sort(col_r27); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const col_r27 = ctx.$implicit;
    const ctx_r24 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ctx_r24.getColumnClass(col_r27));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", col_r27.label, " ");
} }
function DatatableComponent_div_4_Template(rf, ctx) { if (rf & 1) {
    const _r32 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 6);
    ɵngcc0.ɵɵelementStart(1, "input", 8);
    ɵngcc0.ɵɵlistener("keyup", function DatatableComponent_div_4_Template_input_keyup_1_listener($event) { ɵngcc0.ɵɵrestoreView(_r32); const col_r30 = ctx.$implicit; const ctx_r31 = ɵngcc0.ɵɵnextContext(); return ctx_r31.filter($event, col_r30.field, $event.target.value); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const col_r30 = ctx.$implicit;
    const ctx_r25 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ctx_r25.getColumnClass(col_r30));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("name", col_r30.field)("ngModel", ctx_r25.search[col_r30.field]);
} }
function DatatableComponent_div_6_div_1_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div");
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const col_r36 = ctx.$implicit;
    const row_r33 = ɵngcc0.ɵɵnextContext().$implicit;
    ɵngcc0.ɵɵattribute("class", col_r36.className + " ui-dt-cell");
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", row_r33[col_r36.field], " ");
} }
function DatatableComponent_div_6_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵtemplate(1, DatatableComponent_div_6_div_1_Template, 2, 2, "div", 10);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r26 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r26.columns);
} }
const _c6 = function (a0) { return { "active disabled": a0 }; };
function PaginationComponent_li_7_Template(rf, ctx) { if (rf & 1) {
    const _r47 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 8);
    ɵngcc0.ɵɵlistener("click", function PaginationComponent_li_7_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r47); const page_r42 = ctx.$implicit; const ctx_r46 = ɵngcc0.ɵɵnextContext(); return ctx_r46.onPageChange(page_r42); });
    ɵngcc0.ɵɵelementStart(1, "a", 9);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const page_r42 = ctx.$implicit;
    const ctx_r39 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(2, _c6, ctx_r39.pageNum === page_r42));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(page_r42);
} }
const _c7 = function (a0) { return { "disabled": a0 }; };
function PaginationComponent_li_8_Template(rf, ctx) { if (rf & 1) {
    const _r49 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 3);
    ɵngcc0.ɵɵlistener("click", function PaginationComponent_li_8_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r49); const ctx_r48 = ɵngcc0.ɵɵnextContext(); return ctx_r48.shiftBoundary(); });
    ɵngcc0.ɵɵelementStart(1, "a", 7);
    ɵngcc0.ɵɵtext(2, "...");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r40 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(1, _c7, ctx_r40.pageNum === ctx_r40.pages.length));
} }
function PaginationComponent_li_9_Template(rf, ctx) { if (rf & 1) {
    const _r51 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 3);
    ɵngcc0.ɵɵlistener("click", function PaginationComponent_li_9_Template_li_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r51); const ctx_r50 = ɵngcc0.ɵɵnextContext(); return ctx_r50.onPageChange(ctx_r50.pageCount); });
    ɵngcc0.ɵɵelementStart(1, "a", 7);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r41 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(2, _c7, ctx_r41.pageNum === ctx_r41.pageCount));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate1(" ", ctx_r41.pageCount, "");
} }
function DatepickerComponent_div_9_th_22_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "th", 67);
    ɵngcc0.ɵɵtext(1);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const day_r55 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(day_r55);
} }
function DatepickerComponent_div_9_tr_24_td_1_Template(rf, ctx) { if (rf & 1) {
    const _r60 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "td", 69);
    ɵngcc0.ɵɵlistener("click", function DatepickerComponent_div_9_tr_24_td_1_Template_td_click_0_listener() { ɵngcc0.ɵɵrestoreView(_r60); const day_r58 = ctx.$implicit; const ctx_r59 = ɵngcc0.ɵɵnextContext(3); return ctx_r59.setDate(day_r58); });
    ɵngcc0.ɵɵelementStart(1, "span");
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const day_r58 = ctx.$implicit;
    const ctx_r57 = ɵngcc0.ɵɵnextContext(3);
    ɵngcc0.ɵɵproperty("ngClass", ctx_r57.getSelectedClass(day_r58));
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵtextInterpolate(day_r58);
} }
function DatepickerComponent_div_9_tr_24_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementStart(0, "tr", 28);
    ɵngcc0.ɵɵtemplate(1, DatepickerComponent_div_9_tr_24_td_1_Template, 3, 2, "td", 68);
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const row_r56 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", row_r56);
} }
function DatepickerComponent_div_9_Template(rf, ctx) { if (rf & 1) {
    const _r62 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "div", 9);
    ɵngcc0.ɵɵelementStart(1, "a", 10);
    ɵngcc0.ɵɵelement(2, "i", 11);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(3, "div", 12);
    ɵngcc0.ɵɵelementStart(4, "i", 13);
    ɵngcc0.ɵɵlistener("click", function DatepickerComponent_div_9_Template_i_click_4_listener() { ɵngcc0.ɵɵrestoreView(_r62); const ctx_r61 = ɵngcc0.ɵɵnextContext(); return ctx_r61.clear(); });
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(5, "div", 14);
    ɵngcc0.ɵɵelementStart(6, "div", 15);
    ɵngcc0.ɵɵelementStart(7, "span", 16);
    ɵngcc0.ɵɵtext(8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(9, "div", 17);
    ɵngcc0.ɵɵelementStart(10, "div", 18);
    ɵngcc0.ɵɵelementStart(11, "div", 19);
    ɵngcc0.ɵɵelementStart(12, "div", 20);
    ɵngcc0.ɵɵelementStart(13, "button", 21);
    ɵngcc0.ɵɵlistener("click", function DatepickerComponent_div_9_Template_button_click_13_listener() { ɵngcc0.ɵɵrestoreView(_r62); const ctx_r63 = ɵngcc0.ɵɵnextContext(); return ctx_r63.prevMonth(); });
    ɵngcc0.ɵɵelement(14, "i", 22);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(15, "button", 23);
    ɵngcc0.ɵɵlistener("click", function DatepickerComponent_div_9_Template_button_click_15_listener() { ɵngcc0.ɵɵrestoreView(_r62); const ctx_r64 = ɵngcc0.ɵɵnextContext(); return ctx_r64.nextMonth(); });
    ɵngcc0.ɵɵelement(16, "i", 24);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(17, "div", 25);
    ɵngcc0.ɵɵtext(18);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(19, "table", 26);
    ɵngcc0.ɵɵelementStart(20, "thead", 27);
    ɵngcc0.ɵɵelementStart(21, "tr", 28);
    ɵngcc0.ɵɵtemplate(22, DatepickerComponent_div_9_th_22_Template, 2, 1, "th", 29);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(23, "tbody", 30);
    ɵngcc0.ɵɵtemplate(24, DatepickerComponent_div_9_tr_24_Template, 2, 1, "tr", 31);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(25, "div", 32);
    ɵngcc0.ɵɵelementStart(26, "div", 33);
    ɵngcc0.ɵɵelementStart(27, "label", 34);
    ɵngcc0.ɵɵelement(28, "input", 35);
    ɵngcc0.ɵɵtext(29, " AM ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(30, "label", 36);
    ɵngcc0.ɵɵelement(31, "input", 37);
    ɵngcc0.ɵɵtext(32, " PM ");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelement(33, "div", 38);
    ɵngcc0.ɵɵelementStart(34, "div", 39);
    ɵngcc0.ɵɵelementStart(35, "div", 40);
    ɵngcc0.ɵɵtext(36, "3");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(37, "div", 41);
    ɵngcc0.ɵɵtext(38, "4");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(39, "div", 42);
    ɵngcc0.ɵɵtext(40, "5");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(41, "div", 43);
    ɵngcc0.ɵɵtext(42, "6");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(43, "div", 44);
    ɵngcc0.ɵɵtext(44, "7");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(45, "div", 45);
    ɵngcc0.ɵɵtext(46, "8");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(47, "div", 46);
    ɵngcc0.ɵɵtext(48, "9");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(49, "div", 47);
    ɵngcc0.ɵɵtext(50, "10");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(51, "div", 48);
    ɵngcc0.ɵɵtext(52, "11");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(53, "div", 49);
    ɵngcc0.ɵɵtext(54, "12");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(55, "div", 50);
    ɵngcc0.ɵɵtext(56, "1");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(57, "div", 51);
    ɵngcc0.ɵɵtext(58, "2");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(59, "div", 52);
    ɵngcc0.ɵɵelement(60, "div", 53);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(61, "div", 54);
    ɵngcc0.ɵɵelementStart(62, "div", 55);
    ɵngcc0.ɵɵtext(63, "15");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(64, "div", 56);
    ɵngcc0.ɵɵtext(65, "20");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(66, "div", 57);
    ɵngcc0.ɵɵtext(67, "25");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(68, "div", 58);
    ɵngcc0.ɵɵtext(69, "30");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(70, "div", 59);
    ɵngcc0.ɵɵtext(71, "35");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(72, "div", 60);
    ɵngcc0.ɵɵtext(73, "40");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(74, "div", 61);
    ɵngcc0.ɵɵtext(75, "45");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(76, "div", 62);
    ɵngcc0.ɵɵtext(77, "50");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(78, "div", 63);
    ɵngcc0.ɵɵtext(79, "55");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(80, "div", 49);
    ɵngcc0.ɵɵtext(81, "0");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(82, "div", 42);
    ɵngcc0.ɵɵtext(83, "5");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(84, "div", 47);
    ɵngcc0.ɵɵtext(85, "10");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(86, "div", 52);
    ɵngcc0.ɵɵelement(87, "div", 53);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(88, "div", 64);
    ɵngcc0.ɵɵelementStart(89, "a", 65);
    ɵngcc0.ɵɵlistener("click", function DatepickerComponent_div_9_Template_a_click_89_listener() { ɵngcc0.ɵɵrestoreView(_r62); const ctx_r65 = ɵngcc0.ɵɵnextContext(); return ctx_r65.clear(); });
    ɵngcc0.ɵɵtext(90, "Cancel");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(91, "a", 66);
    ɵngcc0.ɵɵlistener("click", function DatepickerComponent_div_9_Template_a_click_91_listener() { ɵngcc0.ɵɵrestoreView(_r62); const ctx_r66 = ɵngcc0.ɵɵnextContext(); return ctx_r66.showCal = false; });
    ɵngcc0.ɵɵtext(92, "OK");
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r52 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵadvance(8);
    ɵngcc0.ɵɵtextInterpolate(ctx_r52.selectedDateTime.formatDate);
    ɵngcc0.ɵɵadvance(10);
    ɵngcc0.ɵɵtextInterpolate2("", ctx_r52.selectedDateTime.monthStringShort, " ", ctx_r52.selectedDateTime.year, "");
    ɵngcc0.ɵɵadvance(4);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r52.defaultLocaleWeekdaysShort);
    ɵngcc0.ɵɵadvance(2);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r52.daysList);
} }
function DropdownComponent_ng_container_4_div_3_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 6);
} }
function DropdownComponent_ng_container_4_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelementContainerStart(0);
    ɵngcc0.ɵɵelementStart(1, "a", 4);
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵtemplate(3, DropdownComponent_ng_container_4_div_3_Template, 1, 0, "div", 5);
    ɵngcc0.ɵɵelementContainerEnd();
} if (rf & 2) {
    const item_r68 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵpropertyInterpolate("href", item_r68.url, ɵngcc0.ɵɵsanitizeUrl);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate(item_r68.text);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngIf", item_r68.isDivider);
} }
function ModalComponent_div_18_Template(rf, ctx) { if (rf & 1) {
    ɵngcc0.ɵɵelement(0, "div", 13);
} }
const _c8 = function (a0) { return { "active show": a0 }; };
function TabComponent_li_1_Template(rf, ctx) { if (rf & 1) {
    const _r76 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 4);
    ɵngcc0.ɵɵelementStart(1, "a", 5);
    ɵngcc0.ɵɵlistener("click", function TabComponent_li_1_Template_a_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r76); const item_r73 = ctx.$implicit; const ctx_r75 = ɵngcc0.ɵɵnextContext(); return ctx_r75.toggle(item_r73.index); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r73 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(4, _c4, item_r73.isOpen));
    ɵngcc0.ɵɵattribute("id", "tab" + item_r73.index)("aria-selected", item_r73.isOpen);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", item_r73.headerText, " ");
} }
function TabComponent_li_2_li_5_Template(rf, ctx) { if (rf & 1) {
    const _r80 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li");
    ɵngcc0.ɵɵelementStart(1, "a", 11);
    ɵngcc0.ɵɵlistener("click", function TabComponent_li_2_li_5_Template_a_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r80); const item_r78 = ctx.$implicit; const ctx_r79 = ɵngcc0.ɵɵnextContext(2); return ctx_r79.toggle(item_r78.index); });
    ɵngcc0.ɵɵtext(2);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const item_r78 = ctx.$implicit;
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(4, _c4, item_r78.isOpen));
    ɵngcc0.ɵɵattribute("id", "tab" + item_r78.index)("aria-selected", item_r78.isOpen);
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵtextInterpolate1(" ", item_r78.headerText, "");
} }
function TabComponent_li_2_Template(rf, ctx) { if (rf & 1) {
    const _r82 = ɵngcc0.ɵɵgetCurrentView();
    ɵngcc0.ɵɵelementStart(0, "li", 6);
    ɵngcc0.ɵɵelementStart(1, "a", 7);
    ɵngcc0.ɵɵlistener("click", function TabComponent_li_2_Template_a_click_1_listener() { ɵngcc0.ɵɵrestoreView(_r82); const ctx_r81 = ɵngcc0.ɵɵnextContext(); return ctx_r81.toggleDropdown(); });
    ɵngcc0.ɵɵtext(2, "more ");
    ɵngcc0.ɵɵelement(3, "span", 8);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementStart(4, "ul", 9);
    ɵngcc0.ɵɵtemplate(5, TabComponent_li_2_li_5_Template, 3, 6, "li", 10);
    ɵngcc0.ɵɵelementEnd();
    ɵngcc0.ɵɵelementEnd();
} if (rf & 2) {
    const ctx_r72 = ɵngcc0.ɵɵnextContext();
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(4, _c2, ctx_r72.isShowDropdown));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵattribute("aria-expanded", ctx_r72.isShowDropdown);
    ɵngcc0.ɵɵadvance(3);
    ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(6, _c2, ctx_r72.isShowDropdown));
    ɵngcc0.ɵɵadvance(1);
    ɵngcc0.ɵɵproperty("ngForOf", ctx_r72.optionTabs);
} }
class NgUikitComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
NgUikitComponent.ɵfac = function NgUikitComponent_Factory(t) { return new (t || NgUikitComponent)(); };
NgUikitComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: NgUikitComponent, selectors: [["lib-ng-uikit"]], decls: 2, vars: 0, template: function NgUikitComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "p");
        ɵngcc0.ɵɵtext(1, " ng-uikit works! ");
        ɵngcc0.ɵɵelementEnd();
    } }, encapsulation: 2 });
NgUikitComponent.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgUikitComponent, [{
        type: Component,
        args: [{
                selector: 'lib-ng-uikit',
                template: `
    <p>
      ng-uikit works!
    </p>
  `,
                styles: []
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class NgUikitModule {
}
NgUikitModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: NgUikitModule });
NgUikitModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function NgUikitModule_Factory(t) { return new (t || NgUikitModule)(); }, imports: [[]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(NgUikitModule, { declarations: [NgUikitComponent] }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(NgUikitModule, [{
        type: NgModule,
        args: [{
                imports: [],
                declarations: [NgUikitComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AccordionComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
AccordionComponent.ɵfac = function AccordionComponent_Factory(t) { return new (t || AccordionComponent)(); };
AccordionComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AccordionComponent, selectors: [["ui-accordion"]], ngContentSelectors: _c0, decls: 2, vars: 0, consts: [["id", "accordionExample", 1, "accordion"]], template: function AccordionComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } }, styles: [""] });
AccordionComponent.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AccordionComponent, [{
        type: Component,
        args: [{
                selector: 'ui-accordion',
                template: `<div class="accordion" id="accordionExample">
  <ng-content></ng-content>
</div>
`,
                styles: [``]
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AccordionPanelHeaderComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
AccordionPanelHeaderComponent.ɵfac = function AccordionPanelHeaderComponent_Factory(t) { return new (t || AccordionPanelHeaderComponent)(); };
AccordionPanelHeaderComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AccordionPanelHeaderComponent, selectors: [["ui-accordion-panel-header"]], ngContentSelectors: _c0, decls: 1, vars: 0, template: function AccordionPanelHeaderComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵprojection(0);
    } }, styles: [""] });
AccordionPanelHeaderComponent.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AccordionPanelHeaderComponent, [{
        type: Component,
        args: [{
                selector: 'ui-accordion-panel-header',
                template: `<ng-content></ng-content>
`,
                styles: [``]
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AccordionPanelComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    get hasHeader() {
        return this.headerComponent && this.headerComponent.length > 0;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.index = Math.ceil(1000 * Math.random());
    }
    /**
     * @param {?} i
     * @return {?}
     */
    toggle(i) {
        this.isOpen = !this.isOpen;
    }
}
AccordionPanelComponent.ɵfac = function AccordionPanelComponent_Factory(t) { return new (t || AccordionPanelComponent)(); };
AccordionPanelComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AccordionPanelComponent, selectors: [["ui-accordion-panel"]], contentQueries: function AccordionPanelComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, AccordionPanelHeaderComponent, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.headerComponent = _t);
    } }, inputs: { isOpen: "isOpen", header: "header" }, ngContentSelectors: _c3, decls: 9, vars: 11, consts: [[1, "card"], [1, "card-header"], [1, "mb-0"], ["data-toggle", "collapse", 3, "click"], ["class", "ui-accordion-header-text", 4, "ngIf"], [4, "ngIf"], ["data-parent", "#accordionExample", 1, "collapse", 3, "ngClass"], [1, "card-body"], [1, "ui-accordion-header-text"]], template: function AccordionPanelComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef(_c1);
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵelementStart(2, "h5", 2);
        ɵngcc0.ɵɵelementStart(3, "a", 3);
        ɵngcc0.ɵɵlistener("click", function AccordionPanelComponent_Template_a_click_3_listener() { return ctx.toggle(ctx.index); });
        ɵngcc0.ɵɵtemplate(4, AccordionPanelComponent_span_4_Template, 2, 1, "span", 4);
        ɵngcc0.ɵɵtemplate(5, AccordionPanelComponent_5_Template, 1, 0, undefined, 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(6, "div", 6);
        ɵngcc0.ɵɵelementStart(7, "div", 7);
        ɵngcc0.ɵɵprojection(8);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("id", "heading" + ctx.index);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵattribute("data-target", "#collapse" + ctx.index)("aria-expanded", ctx.isOpen)("aria-controls", "collapse" + ctx.index);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.hasHeader);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.hasHeader);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(9, _c2, ctx.isOpen));
        ɵngcc0.ɵɵattribute("id", "collapse" + ctx.index)("aria-labelledby", "heading" + ctx.index);
    } }, directives: [ɵngcc1.NgIf, ɵngcc1.NgClass], styles: [""] });
AccordionPanelComponent.ctorParameters = () => [];
AccordionPanelComponent.propDecorators = {
    header: [{ type: Input }],
    isOpen: [{ type: Input }],
    headerComponent: [{ type: ContentChildren, args: [AccordionPanelHeaderComponent,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AccordionPanelComponent, [{
        type: Component,
        args: [{
                selector: 'ui-accordion-panel',
                template: `<div class="card">
  <div class="card-header" [attr.id]="'heading'+index">
    <h5 class="mb-0">
      <a     data-toggle="collapse" [attr.data-target]="'#collapse'+index"
              [attr.aria-expanded]="isOpen" [attr.aria-controls]="'collapse'+index" (click)="toggle(index)">
        <span class="ui-accordion-header-text" *ngIf="!hasHeader">
          {{header}}
        </span>
        <ng-content select="ui-accordion-panel-header" *ngIf="hasHeader"></ng-content>
      </a>
    </h5>
  </div>
  <div [attr.id]="'collapse'+index" class="collapse " [ngClass]="{'show':isOpen}"
       [attr.aria-labelledby]="'heading'+index" data-parent="#accordionExample">
    <div class="card-body">
      <ng-content></ng-content>
    </div>
  </div>
</div>
`,
                styles: [``]
            }]
    }], function () { return []; }, { isOpen: [{
            type: Input
        }], header: [{
            type: Input
        }], headerComponent: [{
            type: ContentChildren,
            args: [AccordionPanelHeaderComponent]
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AccordionModule {
}
AccordionModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AccordionModule });
AccordionModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AccordionModule_Factory(t) { return new (t || AccordionModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AccordionModule, { declarations: function () { return [AccordionComponent,
        AccordionPanelComponent,
        AccordionPanelHeaderComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [AccordionComponent,
        AccordionPanelComponent,
        AccordionPanelHeaderComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AccordionModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [AccordionComponent, AccordionPanelComponent, AccordionPanelHeaderComponent],
                exports: [AccordionComponent, AccordionPanelComponent, AccordionPanelHeaderComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AutocompleteComponent {
    /**
     * @param {?} http
     */
    constructor(http) {
        this.http = http;
        this.label = 'Name';
        this.selected = new EventEmitter();
        this.helpText = 'Enter at least 2 char to search';
        this.filteredList = [];
        this.selectedItem = (this.niddle) ? {} : "";
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        if (this.sourceURL) {
            this.loadResponse();
        }
    }
    /**
     * @param {?} key
     * @return {?}
     */
    filter(key) {
        if (key && key.length >= 2) {
            this.filteredList = this.filterResult(key);
            console.log(this.filteredList);
        }
        else {
            this.filteredList = [];
            this.selectedItem = (this.niddle) ? {} : "";
        }
    }
    /**
     * @param {?} key
     * @return {?}
     */
    filterResult(key) {
        /** @type {?} */
        const result = [];
        this.source.forEach((item) => {
            /** @type {?} */
            const name = (this.niddle) ? item[this.niddle].toLowerCase() : item.toString().toLowerCase();
            if (name.indexOf(key.toLowerCase()) !== -1 && (result.length < 10)) {
                result.push(item);
            }
            else {
                return true;
            }
        });
        return result;
    }
    /**
     * @return {?}
     */
    loadResponse() {
        /** @type {?} */
        const url = this.sourceURL || '../../assets/autocomplete.json';
        this.http.get(url).subscribe((res) => {
            if (Array.isArray(res)) {
                this.source = res;
            }
        });
    }
    /**
     * @param {?} item
     * @return {?}
     */
    getText(item) {
        return (this.niddle) ? item[this.niddle] : item;
    }
    /**
     * @param {?} item
     * @return {?}
     */
    selectItem(item) {
        this.selectedItem = item;
        this.filteredList = [];
        this.selected.emit(item);
    }
}
AutocompleteComponent.ɵfac = function AutocompleteComponent_Factory(t) { return new (t || AutocompleteComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc2.HttpClient)); };
AutocompleteComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: AutocompleteComponent, selectors: [["ui-autocomplete"]], inputs: { label: "label", helpText: "helpText", source: "source", sourceURL: "sourceURL", niddle: "niddle" }, outputs: { selected: "selected" }, decls: 8, vars: 5, consts: [[1, "form-group", "ui-autocomplete"], ["for", "UIAutocomplete"], ["type", "text", "class", "form-control", "id", "UIAutocomplete1", "aria-describedby", "emailHelp", "autocomplete", "off", 3, "placeholder", "ngModel", "keyup", 4, "ngIf"], ["type", "text", "class", "form-control", "id", "UIAutocomplete", "aria-describedby", "emailHelp", "autocomplete", "off", 3, "placeholder", "ngModel", "keyup", 4, "ngIf"], ["id", "emailHelp", "class", "form-text text-muted", 4, "ngIf"], [1, "ui-autocomplete-container"], ["class", "list-group ui-autocomplete-list", 4, "ngIf"], ["type", "text", "id", "UIAutocomplete1", "aria-describedby", "emailHelp", "autocomplete", "off", 1, "form-control", 3, "placeholder", "ngModel", "keyup"], ["type", "text", "id", "UIAutocomplete", "aria-describedby", "emailHelp", "autocomplete", "off", 1, "form-control", 3, "placeholder", "ngModel", "keyup"], ["id", "emailHelp", 1, "form-text", "text-muted"], [1, "list-group", "ui-autocomplete-list"], ["class", "list-group-item ui-autocomplete-list-item", 3, "click", 4, "ngFor", "ngForOf"], [1, "list-group-item", "ui-autocomplete-list-item", 3, "click"]], template: function AutocompleteComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "label", 1);
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(3, AutocompleteComponent_input_3_Template, 1, 2, "input", 2);
        ɵngcc0.ɵɵtemplate(4, AutocompleteComponent_input_4_Template, 1, 2, "input", 3);
        ɵngcc0.ɵɵtemplate(5, AutocompleteComponent_small_5_Template, 2, 1, "small", 4);
        ɵngcc0.ɵɵelementStart(6, "div", 5);
        ɵngcc0.ɵɵtemplate(7, AutocompleteComponent_ul_7_Template, 2, 1, "ul", 6);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.label);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.niddle);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", !ctx.niddle);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.helpText);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.filteredList.length);
    } }, directives: [ɵngcc1.NgIf, ɵngcc3.DefaultValueAccessor, ɵngcc3.NgControlStatus, ɵngcc3.NgModel, ɵngcc1.NgForOf], styles: [".ui-autocomplete-container[_ngcontent-%COMP%]{position:relative}.ui-autocomplete-list[_ngcontent-%COMP%]{position:absolute;z-index:9;width:100%}.ui-autocomplete-list-item[_ngcontent-%COMP%]:hover{cursor:pointer;color:#fff;background-color:#007bff;border-color:#007bff}"] });
AutocompleteComponent.ctorParameters = () => [
    { type: HttpClient }
];
AutocompleteComponent.propDecorators = {
    source: [{ type: Input }],
    sourceURL: [{ type: Input }],
    niddle: [{ type: Input }],
    label: [{ type: Input }],
    selected: [{ type: Output }],
    helpText: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AutocompleteComponent, [{
        type: Component,
        args: [{
                selector: 'ui-autocomplete',
                template: `<div class="form-group ui-autocomplete">
  <label for="UIAutocomplete">{{label}}</label>
  <input *ngIf="niddle" type="text" class="form-control" id="UIAutocomplete1" aria-describedby="emailHelp" placeholder="Enter {{label}}"
         [ngModel]="selectedItem[niddle]" (keyup)="filter($event.target.value)" autocomplete="off">

  <input *ngIf="!niddle" type="text" class="form-control" id="UIAutocomplete" aria-describedby="emailHelp" placeholder="Enter {{label}}"
         [ngModel]="selectedItem" (keyup)="filter($event.target.value)" autocomplete="off">
  <small id="emailHelp" class="form-text text-muted" *ngIf="helpText">{{helpText}}</small>
  <div class="ui-autocomplete-container">
    <ul class="list-group ui-autocomplete-list" *ngIf="filteredList.length">
      <li class="list-group-item ui-autocomplete-list-item" (click)="selectItem(item)" *ngFor="let item of filteredList">{{getText(item)}}</li>
    </ul>
  </div>
</div>
`,
                styles: [`.ui-autocomplete-container{position:relative}.ui-autocomplete-list{position:absolute;z-index:9;width:100%}.ui-autocomplete-list-item:hover{cursor:pointer;color:#fff;background-color:#007bff;border-color:#007bff}`]
            }]
    }], function () { return [{ type: ɵngcc2.HttpClient }]; }, { label: [{
            type: Input
        }], selected: [{
            type: Output
        }], helpText: [{
            type: Input
        }], source: [{
            type: Input
        }], sourceURL: [{
            type: Input
        }], niddle: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class AutocompleteModule {
}
AutocompleteModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: AutocompleteModule });
AutocompleteModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function AutocompleteModule_Factory(t) { return new (t || AutocompleteModule)(); }, imports: [[
            CommonModule,
            FormsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(AutocompleteModule, { declarations: function () { return [AutocompleteComponent]; }, imports: function () { return [CommonModule,
        FormsModule]; }, exports: function () { return [AutocompleteComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(AutocompleteModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule
                ],
                declarations: [AutocompleteComponent],
                exports: [AutocompleteComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class CardComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
CardComponent.ɵfac = function CardComponent_Factory(t) { return new (t || CardComponent)(); };
CardComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CardComponent, selectors: [["ui-card"]], inputs: { config: "config" }, ngContentSelectors: _c0, decls: 1, vars: 1, consts: [["class", "card", "style", "width: 18rem;", 4, "ngIf"], [1, "card", 2, "width", "18rem"], ["class", "card-img-top", 3, "src", "alt", 4, "ngIf"], [1, "card-body"], [1, "card-title"], [1, "card-img-top", 3, "src", "alt"]], template: function CardComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵtemplate(0, CardComponent_div_0_Template, 6, 2, "div", 0);
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngIf", ctx.config);
    } }, directives: [ɵngcc1.NgIf], styles: [""] });
CardComponent.ctorParameters = () => [];
CardComponent.propDecorators = {
    config: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CardComponent, [{
        type: Component,
        args: [{
                selector: 'ui-card',
                template: `<div class="card" style="width: 18rem;" *ngIf="config">
  <img class="card-img-top" src="{{config.src}}" alt="{{config.alt || config.title}}" *ngIf="config.src">
  <div class="card-body">
    <h5 class="card-title">{{config.title}}</h5>
    <ng-content></ng-content>
  </div>
</div>
`,
                styles: [``]
            }]
    }], function () { return []; }, { config: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class CardModule {
}
CardModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CardModule });
CardModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CardModule_Factory(t) { return new (t || CardModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CardModule, { declarations: function () { return [CardComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [CardComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CardModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [CardComponent],
                exports: [CardComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class CarouselComponent {
    constructor() {
        this.items = [];
        this.afterNext = new EventEmitter();
        this.afterPrev = new EventEmitter();
        this.width = '600px';
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.activeIndex = 0;
        this.count = this.items.length;
    }
    /**
     * @return {?}
     */
    next() {
        (this.activeIndex < this.count - 1) ? this.activeIndex++ : this.activeIndex = 0;
        this.afterNext.emit({
            index: this.activeIndex,
            item: this.items[this.activeIndex]
        });
    }
    /**
     * @return {?}
     */
    prev() {
        (this.activeIndex >= 1) ? this.activeIndex-- : this.activeIndex = this.count - 1;
        this.afterPrev.emit({
            index: this.activeIndex,
            item: this.items[this.activeIndex]
        });
    }
    /**
     * @param {?} i
     * @return {?}
     */
    goTo(i) {
        this.activeIndex = i;
    }
}
CarouselComponent.ɵfac = function CarouselComponent_Factory(t) { return new (t || CarouselComponent)(); };
CarouselComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: CarouselComponent, selectors: [["ui-carousel"]], inputs: { items: "items" }, outputs: { afterNext: "afterNext", afterPrev: "afterPrev" }, decls: 13, vars: 5, consts: [["id", "carouselExampleIndicators", "data-ride", "carousel", 1, "carousel", "slide", 3, "ngStyle"], [1, "carousel-indicators"], ["data-target", "#carouselExampleIndicators", 3, "ngClass", "click", 4, "ngFor", "ngForOf"], [1, "carousel-inner"], ["class", "carousel-item", 3, "ngClass", 4, "ngFor", "ngForOf"], ["role", "button", "data-slide", "prev", 1, "carousel-control-prev", 3, "click"], ["aria-hidden", "true", 1, "carousel-control-prev-icon"], [1, "sr-only"], ["role", "button", "data-slide", "next", 1, "carousel-control-next", 3, "click"], ["aria-hidden", "true", 1, "carousel-control-next-icon"], ["data-target", "#carouselExampleIndicators", 3, "ngClass", "click"], [1, "carousel-item", 3, "ngClass"], [3, "src", "alt"], [1, "carousel-caption", "d-none", "d-md-block"]], template: function CarouselComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "ol", 1);
        ɵngcc0.ɵɵtemplate(2, CarouselComponent_li_2_Template, 1, 4, "li", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 3);
        ɵngcc0.ɵɵtemplate(4, CarouselComponent_div_4_Template, 7, 7, "div", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "a", 5);
        ɵngcc0.ɵɵlistener("click", function CarouselComponent_Template_a_click_5_listener() { return ctx.prev(); });
        ɵngcc0.ɵɵelement(6, "span", 6);
        ɵngcc0.ɵɵelementStart(7, "span", 7);
        ɵngcc0.ɵɵtext(8, "Previous");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(9, "a", 8);
        ɵngcc0.ɵɵlistener("click", function CarouselComponent_Template_a_click_9_listener() { return ctx.next(); });
        ɵngcc0.ɵɵelement(10, "span", 9);
        ɵngcc0.ɵɵelementStart(11, "span", 7);
        ɵngcc0.ɵɵtext(12, "Next");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngStyle", ɵngcc0.ɵɵpureFunction1(3, _c5, ctx.width));
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
    } }, directives: [ɵngcc1.NgStyle, ɵngcc1.NgForOf, ɵngcc1.NgClass], styles: [""] });
CarouselComponent.ctorParameters = () => [];
CarouselComponent.propDecorators = {
    items: [{ type: Input }],
    afterNext: [{ type: Output }],
    afterPrev: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CarouselComponent, [{
        type: Component,
        args: [{
                selector: 'ui-carousel',
                template: `<div id="carouselExampleIndicators" class="carousel slide" data-ride="carousel" [ngStyle]="{width: width}">
  <ol class="carousel-indicators">
      <li data-target="#carouselExampleIndicators" attr.data-slide-to="{{i}}" *ngFor="let item of items; let i=index;"
          [ngClass]="{'active':i == activeIndex}" (click)="goTo(i)">
      </li>
  </ol>
  <div class="carousel-inner">
    <div class="carousel-item" *ngFor="let item of items; let i=index;" [ngClass]="{'active':i == activeIndex}">
      <img src="{{item.src}}" alt="{{item.alt}}">
      <div class="carousel-caption d-none d-md-block">
        <h5>{{item.alt}}</h5>
        <p>{{item.alt}}</p>
      </div>
    </div>
  </div>
  <a class="carousel-control-prev" (click)="prev()" role="button" data-slide="prev">
    <span class="carousel-control-prev-icon" aria-hidden="true"></span>
    <span class="sr-only">Previous</span>
  </a>
  <a class="carousel-control-next" (click)="next()" role="button" data-slide="next">
    <span class="carousel-control-next-icon" aria-hidden="true"></span>
    <span class="sr-only">Next</span>
  </a>
</div>
`,
                styles: [``]
            }]
    }], function () { return []; }, { items: [{
            type: Input
        }], afterNext: [{
            type: Output
        }], afterPrev: [{
            type: Output
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class CarouselModule {
}
CarouselModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: CarouselModule });
CarouselModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function CarouselModule_Factory(t) { return new (t || CarouselModule)(); }, imports: [[
            CommonModule,
            BrowserModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(CarouselModule, { declarations: function () { return [CarouselComponent]; }, imports: function () { return [CommonModule,
        BrowserModule]; }, exports: function () { return [CarouselComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(CarouselModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    BrowserModule
                ],
                declarations: [CarouselComponent],
                exports: [CarouselComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class ChartComponent {
    /**
     * @param {?} el
     */
    constructor(el) {
        this.el = el;
        this.width = '400px';
        this.height = '400px';
        this.clickChart = new EventEmitter();
        this.responsive = true;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
    /**
     * @return {?}
     */
    ngAfterViewInit() {
        /** @type {?} */
        const ctx = this.el.nativeElement.children[0].children[0].getContext('2d');
        /** @type {?} */
        const _that = this;
        Chart.plugins.register({
            afterDatasetsDraw: function (chartInstance, easing) {
                // To only draw at the end of animation, check for easing === 1
                /** @type {?} */
                var ctx = chartInstance.chart.ctx;
                chartInstance.data.datasets.forEach(function (dataset, i) {
                    /** @type {?} */
                    var meta = chartInstance.getDatasetMeta(i);
                    if (!meta.hidden) {
                        meta.data.forEach(function (element, index) {
                            // Draw the text in black, with the specified font
                            ctx.fillStyle = 'rgb(0, 0, 0)';
                            /** @type {?} */
                            var fontSize = 16;
                            /** @type {?} */
                            var fontStyle = 'normal';
                            /** @type {?} */
                            var fontFamily = 'Helvetica Neue';
                            ctx.font = Chart.helpers.fontString(fontSize, fontStyle, fontFamily);
                            // Just naively convert to string for now
                            /** @type {?} */
                            var dataString = dataset.data[index].toString();
                            // Make sure alignment settings are correct
                            ctx.textAlign = 'center';
                            ctx.textBaseline = 'middle';
                            /** @type {?} */
                            var padding = 5;
                            /** @type {?} */
                            var position = element.tooltipPosition();
                            ctx.fillText(dataString, position.x, position.y - (fontSize / 2) - padding);
                        });
                    }
                });
            }
        });
        this.chart = new Chart(ctx, {
            // The type of chart we want to create
            type: 'bar',
            // The data for our dataset
            data: this.data,
            maintainAspectRatio: false,
            responsive: true,
            // Configuration options go here
            options: {
                onClick: function (e) {
                    /** @type {?} */
                    let activePoints = _that.chart.getElementsAtEvent(e);
                    /** @type {?} */
                    let index = activePoints[0]._index;
                    _that.clickChart.emit({ y: this.data.datasets[0].data[index], x: this.data.labels[index] });
                },
                title: {
                    display: true,
                    text: 'My Title',
                    fontSize: 16
                },
                legend: {
                    enabled: false,
                },
                scales: {
                    yAxes: [{
                            ticks: {
                                beginAtZero: true
                            }
                        }]
                },
                layout: {
                    padding: {
                        left: 50,
                        right: 0,
                        top: 0,
                        bottom: 0
                    }
                }
            },
            onClick: function (e) {
                /** @type {?} */
                let activePoints = _that.chart.getElementsAtEvent(e);
                /** @type {?} */
                let selectedIndex = activePoints[0]._index;
                console.log(this.data.datasets[0].data[selectedIndex]);
            }
        });
    }
    /**
     * @param {?} e
     * @return {?}
     */
    onCanvasClick(e) {
        // if(this.chart) {
        //   let element = this.chart.getElementAtEvent(event);
        //   let dataset = this.chart.getDatasetAtEvent(event);
        //   if(element && element[0] && dataset) {
        //     this.clickChart.emit({originalEvent: event, element: element[0], dataset: dataset});
        //   }
        // }
    }
}
ChartComponent.ɵfac = function ChartComponent_Factory(t) { return new (t || ChartComponent)(ɵngcc0.ɵɵdirectiveInject(ɵngcc0.ElementRef)); };
ChartComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ChartComponent, selectors: [["ui-chart"]], inputs: { width: "width", height: "height", data: "data" }, outputs: { clickChart: "clickChart" }, decls: 2, vars: 6, consts: [[2, "position", "relative"], [3, "click"]], template: function ChartComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "canvas", 1);
        ɵngcc0.ɵɵlistener("click", function ChartComponent_Template_canvas_click_1_listener($event) { return ctx.onCanvasClick($event); });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵstyleProp("width", ctx.responsive && !ctx.width ? null : ctx.width)("height", ctx.responsive && !ctx.height ? null : ctx.height);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("width", ctx.responsive && !ctx.width ? null : ctx.width)("height", ctx.responsive && !ctx.height ? null : ctx.height);
    } }, styles: [""] });
ChartComponent.ctorParameters = () => [
    { type: ElementRef }
];
ChartComponent.propDecorators = {
    data: [{ type: Input }],
    width: [{ type: Input }],
    height: [{ type: Input }],
    clickChart: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChartComponent, [{
        type: Component,
        args: [{
                selector: 'ui-chart',
                template: `<div style="position:relative" [style.width]="responsive && !width ? null : width" [style.height]="responsive && !height ? null : height">
  <canvas [attr.width]="responsive && !width ? null : width" [attr.height]="responsive && !height ? null : height" (click)="onCanvasClick($event)"></canvas>
</div>
`,
                styles: [``]
            }]
    }], function () { return [{ type: ɵngcc0.ElementRef }]; }, { width: [{
            type: Input
        }], height: [{
            type: Input
        }], clickChart: [{
            type: Output
        }], data: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class ChartModule {
}
ChartModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ChartModule });
ChartModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ChartModule_Factory(t) { return new (t || ChartModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ChartModule, { declarations: function () { return [ChartComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ChartComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ChartModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [ChartComponent],
                exports: [ChartComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DatatableService {
    constructor() { }
    /**
     * @param {?} rows
     * @param {?} size
     * @param {?} pageNum
     * @return {?}
     */
    getPageData(rows, size, pageNum) {
        /** @type {?} */
        const start = (pageNum - 1) * size;
        /** @type {?} */
        const last = (start + size) > rows.length ? rows.length : start + size;
        return {
            startIndex: start,
            rows: rows.slice(start, last),
            lastIndex: last
        };
    }
}
DatatableService.ɵfac = function DatatableService_Factory(t) { return new (t || DatatableService)(); };
DatatableService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: DatatableService, factory: DatatableService.ɵfac, providedIn: 'root' });
DatatableService.ctorParameters = () => [];
/** @nocollapse */ DatatableService.ngInjectableDef = defineInjectable({ factory: function DatatableService_Factory() { return new DatatableService(); }, token: DatatableService, providedIn: "root" });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DatatableService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DatatableComponent {
    /**
     * @param {?} _dts
     */
    constructor(_dts) {
        this._dts = _dts;
        this.pageChange = new EventEmitter();
        this.sorted = new EventEmitter();
        this.size = 6;
        this.search = {};
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.page = this._dts.getPageData(this.rows, this.size, 1);
        console.log(this.page);
        this.resultCount = this.rows.length;
    }
    /**
     * @param {?} col
     * @return {?}
     */
    getColumnClass(col) {
        /** @type {?} */
        let className = ' ui-dt-cell ';
        if (col.className) {
            className += col.className + ' ';
        }
        if (col.sortable) {
            className += ' sortable';
        }
        return className;
    }
    /**
     * @param {?} e
     * @return {?}
     */
    onPageChange(e) {
        this.page = this._dts.getPageData(this.rows, this.size, e.pageNum);
    }
    /**
     * @param {?} item
     * @return {?}
     */
    sort(item) {
        this.page.rows.sort(this.dynamicSort(item.field));
    }
    /**
     * @param {?} property
     * @return {?}
     */
    dynamicSort(property) {
        /** @type {?} */
        var sortOrder = 1;
        if (property[0] === '-') {
            sortOrder = -1;
            property = property.substr(1);
        }
        return function (a, b) {
            if (sortOrder === -1) {
                return b[property].toString().localeCompare(a[property]);
            }
            else {
                return a[property].toString().localeCompare(b[property]);
            }
        };
    }
    /**
     * @param {?} event
     * @param {?} field
     * @param {?} val
     * @return {?}
     */
    filter(event, field, val) {
        if (val.toString().length) {
            if (event.keyCode === 13 || event.which === 13) {
                this.search = {};
                this.search[field] = val;
                console.log(field, this.search, this.rows);
                /** @type {?} */
                const rows = [];
                this.rows.forEach((item) => {
                    if (item[field].toString().toLowerCase().indexOf(val.toString().toLowerCase()) !== -1) {
                        rows.push(item);
                    }
                });
                this.resultCount = rows.length;
                this.page = this._dts.getPageData(rows, this.size, 1);
            }
        }
        else {
            this.resultCount = this.rows.length;
            this.page = this._dts.getPageData(this.rows, this.size, 1);
        }
    }
}
DatatableComponent.ɵfac = function DatatableComponent_Factory(t) { return new (t || DatatableComponent)(ɵngcc0.ɵɵdirectiveInject(DatatableService)); };
DatatableComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DatatableComponent, selectors: [["ui-datatable"]], inputs: { size: "size", rows: "rows", columns: "columns" }, outputs: { pageChange: "pageChange", sorted: "sorted" }, decls: 8, vars: 5, consts: [[1, "ui-dt-table"], [1, "ui-dt-thead", "row"], [3, "ngClass", 4, "ngFor", "ngForOf"], [1, "ui-dt-tbody"], ["class", "row ui-dt-row", 4, "ngFor", "ngForOf"], [3, "size", "totalCount", "pageChange"], [3, "ngClass"], [1, "fas", "fa-sort-amount-up", 3, "click"], ["type", "text", 1, "form-control", "form-control-sm", 3, "name", "ngModel", "keyup"], [1, "row", "ui-dt-row"], [4, "ngFor", "ngForOf"]], template: function DatatableComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "div", 1);
        ɵngcc0.ɵɵtemplate(2, DatatableComponent_div_2_Template, 3, 2, "div", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 1);
        ɵngcc0.ɵɵtemplate(4, DatatableComponent_div_4_Template, 2, 3, "div", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(5, "div", 3);
        ɵngcc0.ɵɵtemplate(6, DatatableComponent_div_6_Template, 2, 1, "div", 4);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "ui-pagination", 5);
        ɵngcc0.ɵɵlistener("pageChange", function DatatableComponent_Template_ui_pagination_pageChange_7_listener($event) { return ctx.onPageChange($event); });
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.columns);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.columns);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.page.rows);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("size", ctx.size)("totalCount", ctx.resultCount);
    } }, directives: function () { return [ɵngcc1.NgForOf, PaginationComponent, ɵngcc1.NgClass, ɵngcc3.DefaultValueAccessor, ɵngcc3.NgControlStatus, ɵngcc3.NgModel]; }, styles: [".ui-dt-thead[_ngcontent-%COMP%]{border-bottom:1px solid #999}.ui-dt-thead[_ngcontent-%COMP%]   .ui-dt-cell[_ngcontent-%COMP%]{font-weight:500}.ui-dt-row[_ngcontent-%COMP%]{border-bottom:1px solid #999}.ui-dt-cell[_ngcontent-%COMP%]{-ms-word-break:break-word;word-break:break-word;border-left:1px solid #ddd;padding:5px}.ui-dt-row[_ngcontent-%COMP%]   .ui-dt-cell[_ngcontent-%COMP%]:last-child{border-right:1px solid #ddd}"] });
DatatableComponent.ctorParameters = () => [
    { type: DatatableService }
];
DatatableComponent.propDecorators = {
    rows: [{ type: Input }],
    columns: [{ type: Input }],
    pageChange: [{ type: Output }],
    sorted: [{ type: Output }],
    size: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DatatableComponent, [{
        type: Component,
        args: [{
                selector: 'ui-datatable',
                template: `<div class="ui-dt-table">
  <div class="ui-dt-thead row">
    <div *ngFor="let col of columns" [ngClass]="getColumnClass(col)"  >
      {{col.label}}
      <i class="fas fa-sort-amount-up" (click)="sort(col)"></i>
    </div>
  </div>
  <div class="ui-dt-thead row">
    <div *ngFor="let col of columns" [ngClass]="getColumnClass(col)"  >
        <input type="text" class="form-control form-control-sm"
               [name]="col.field"
               [ngModel]="search[col.field]"
               (keyup)="filter($event, col.field, $event.target.value)"
        >
    </div>
  </div>
  <div class="ui-dt-tbody">
    <div *ngFor="let row of page.rows;let i=index;" class="row ui-dt-row">
      <div [attr.class]="col.className +' ui-dt-cell'" *ngFor="let col of columns;let k=index;">
         {{row[col.field]}}
      </div>
    </div>
  </div>
</div>
<ui-pagination
  [size]="size"
  [totalCount]="resultCount"
  (pageChange)="onPageChange($event)">
</ui-pagination>

`,
                styles: [`.ui-dt-thead{border-bottom:1px solid #999}.ui-dt-thead .ui-dt-cell{font-weight:500}.ui-dt-row{border-bottom:1px solid #999}.ui-dt-cell{-ms-word-break:break-word;word-break:break-word;border-left:1px solid #ddd;padding:5px}.ui-dt-row .ui-dt-cell:last-child{border-right:1px solid #ddd}`]
            }]
    }], function () { return [{ type: DatatableService }]; }, { pageChange: [{
            type: Output
        }], sorted: [{
            type: Output
        }], size: [{
            type: Input
        }], rows: [{
            type: Input
        }], columns: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class PagingService {
    constructor() {
    }
    /**
     * @return {?}
     */
    get size() {
        return this._size;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set size(value) {
        this._size = value;
    }
    /**
     * @return {?}
     */
    get totalCount() {
        return this._totalCount;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set totalCount(value) {
        this._totalCount = value;
    }
    /**
     * @param {?} totalCount
     * @return {?}
     */
    getListOfPage(totalCount) {
        return Math.ceil(totalCount / this._size);
    }
}
PagingService.ɵfac = function PagingService_Factory(t) { return new (t || PagingService)(); };
PagingService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: PagingService, factory: PagingService.ɵfac, providedIn: 'root' });
PagingService.ctorParameters = () => [];
/** @nocollapse */ PagingService.ngInjectableDef = defineInjectable({ factory: function PagingService_Factory() { return new PagingService(); }, token: PagingService, providedIn: "root" });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PagingService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class PaginationComponent {
    /**
     * @param {?} _pgs
     */
    constructor(_pgs) {
        this._pgs = _pgs;
        this.size = 10;
        this.pageChange = new EventEmitter();
        this.pageNum = 1;
        this.displayNumbers = 5;
        this.lastPage = 5;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this._pgs.size = this.size;
        this._pgs.totalCount = this.totalCount;
        this.pageCount = this._pgs.getListOfPage(this.totalCount);
        this.totalPages = Array(this.pageCount).fill(0).map((e, i) => i + 1);
        this.pages = this.totalPages.slice(0, 5);
        this.lastPage = this.pages[this.displayNumbers - 1];
    }
    /**
     * @param {?} e
     * @return {?}
     */
    onPageChange(e) {
        console.log(e);
        this.pageNum = e;
        this.pageChange.emit({
            pageNum: e
        });
    }
    /**
     * @param {?=} isReverse
     * @return {?}
     */
    shiftBoundary(isReverse = false) {
        /** @type {?} */
        let startIndex = (isReverse) ? this.pageNum - this.displayNumbers : this.pageNum - 1;
        /** @type {?} */
        let lastIndex = (isReverse) ? this.pageNum : this.pageNum + this.displayNumbers - 1;
        this.pages = this.totalPages.slice(startIndex, lastIndex);
        this.lastPage = lastIndex;
    }
    /**
     * @return {?}
     */
    next() {
        this.pageNum++;
        if (this.pageNum > this.lastPage)
            this.shiftBoundary();
    }
    /**
     * @return {?}
     */
    prev() {
        this.pageNum--;
        if ((this.pageNum) % this.displayNumbers === 0) {
            this.shiftBoundary(true);
        }
    }
}
PaginationComponent.ɵfac = function PaginationComponent_Factory(t) { return new (t || PaginationComponent)(ɵngcc0.ɵɵdirectiveInject(PagingService)); };
PaginationComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: PaginationComponent, selectors: [["ui-pagination"]], inputs: { size: "size", totalCount: "totalCount" }, outputs: { pageChange: "pageChange" }, decls: 13, vars: 11, consts: [[1, "m-3"], ["aria-label", "Page navigation example"], [1, "pagination", "justify-content-center"], [1, "page-item", 3, "ngClass", "click"], ["tabindex", "-1", 1, "page-link"], ["class", "page-item", 3, "click", 4, "ngFor", "ngForOf"], ["class", "page-item", 3, "ngClass", "click", 4, "ngIf"], [1, "page-link"], [1, "page-item", 3, "click"], [1, "page-link", 3, "ngClass"]], template: function PaginationComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "nav", 1);
        ɵngcc0.ɵɵelementStart(2, "ul", 2);
        ɵngcc0.ɵɵelementContainerStart(3);
        ɵngcc0.ɵɵelementStart(4, "li", 3);
        ɵngcc0.ɵɵlistener("click", function PaginationComponent_Template_li_click_4_listener() { return ctx.prev(); });
        ɵngcc0.ɵɵelementStart(5, "a", 4);
        ɵngcc0.ɵɵtext(6, "Previous ");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(7, PaginationComponent_li_7_Template, 3, 4, "li", 5);
        ɵngcc0.ɵɵtemplate(8, PaginationComponent_li_8_Template, 3, 3, "li", 6);
        ɵngcc0.ɵɵtemplate(9, PaginationComponent_li_9_Template, 3, 4, "li", 6);
        ɵngcc0.ɵɵelementStart(10, "li", 3);
        ɵngcc0.ɵɵlistener("click", function PaginationComponent_Template_li_click_10_listener() { return ctx.next(); });
        ɵngcc0.ɵɵelementStart(11, "a", 7);
        ɵngcc0.ɵɵtext(12, " Next");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementContainerEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(4);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(7, _c7, ctx.pageNum === 1));
        ɵngcc0.ɵɵattribute("disabled", ctx.pageNum === 1);
        ɵngcc0.ɵɵadvance(3);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.pages);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.pageCount > ctx.displayNumbers && ctx.lastPage < ctx.pageCount);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.pageCount > ctx.displayNumbers && ctx.lastPage < ctx.pageCount);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(9, _c7, ctx.pageNum === ctx.pageCount));
        ɵngcc0.ɵɵattribute("disabled", ctx.pageNum === ctx.pageCount ? "disabled" : "");
    } }, directives: [ɵngcc1.NgClass, ɵngcc1.NgForOf, ɵngcc1.NgIf], styles: [".page-link[_ngcontent-%COMP%]:not([href]):not([tabindex]){position:relative;display:block;padding:.5rem .75rem;margin-left:-1px;line-height:1.25;color:#337ab7;background-color:#fff;border:1px solid #dee2e6}.page-link.active[_ngcontent-%COMP%]:not([href]):not([tabindex]){position:relative;display:block;padding:.5rem .75rem;margin-left:-1px;line-height:1.25;color:#fff;background-color:#337ab7;border:1px solid #337ab7}.disabled[_ngcontent-%COMP%]{pointer-events:none}"] });
PaginationComponent.ctorParameters = () => [
    { type: PagingService }
];
PaginationComponent.propDecorators = {
    totalCount: [{ type: Input }],
    size: [{ type: Input }],
    pageChange: [{ type: Output }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(PaginationComponent, [{
        type: Component,
        args: [{
                selector: 'ui-pagination',
                template: `<div class="m-3">
  <nav aria-label="Page navigation example">
    <ul class="pagination justify-content-center">
      <ng-container>

        <li class="page-item " [ngClass]="{'disabled':pageNum === 1}" (click)="prev()" [attr.disabled]="pageNum === 1">
          <a class="page-link" tabindex="-1">Previous </a>
        </li>

        <li class="page-item" (click)="onPageChange(page)" *ngFor="let page of pages; index as i; first as isFirst; last as isLast;">
          <a class="page-link" [ngClass]="{'active disabled':pageNum===page}">{{page}}</a>
        </li>

        <li class="page-item" *ngIf=" pageCount > displayNumbers && lastPage < pageCount" [ngClass]="{'disabled':pageNum===pages.length}" (click)="shiftBoundary()">
          <a class="page-link">...</a>
        </li>

        <li class="page-item" *ngIf="pageCount > displayNumbers && lastPage < pageCount" [ngClass]="{'disabled':pageNum===pageCount}" (click)="onPageChange(pageCount)">
          <a class="page-link"> {{pageCount}}</a>
        </li>

        <li class="page-item" [ngClass]="{'disabled':pageNum === pageCount}" [attr.disabled]="(pageNum === pageCount)?'disabled':''" (click)="next()">
          <a class="page-link"> Next</a>
        </li>
      </ng-container>
    </ul>
  </nav>
</div>

`,
                styles: [`.page-link:not([href]):not([tabindex]){position:relative;display:block;padding:.5rem .75rem;margin-left:-1px;line-height:1.25;color:#337ab7;background-color:#fff;border:1px solid #dee2e6}.page-link.active:not([href]):not([tabindex]){position:relative;display:block;padding:.5rem .75rem;margin-left:-1px;line-height:1.25;color:#fff;background-color:#337ab7;border:1px solid #337ab7}.disabled{pointer-events:none}`]
            }]
    }], function () { return [{ type: PagingService }]; }, { size: [{
            type: Input
        }], pageChange: [{
            type: Output
        }], totalCount: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DatatableModule {
}
DatatableModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DatatableModule });
DatatableModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DatatableModule_Factory(t) { return new (t || DatatableModule)(); }, providers: [DatatableService], imports: [[
            CommonModule,
            FormsModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DatatableModule, { declarations: function () { return [DatatableComponent,
        PaginationComponent]; }, imports: function () { return [CommonModule,
        FormsModule]; }, exports: function () { return [DatatableComponent,
        PaginationComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DatatableModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    FormsModule
                ],
                providers: [DatatableService],
                declarations: [DatatableComponent, PaginationComponent],
                exports: [DatatableComponent, PaginationComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
/** @type {?} */
const DefaultLocaleMonths = 'January_February_March_April_May_June_July_August_September_October_November_December'.split('_');
/** @type {?} */
const DefaultLocaleMonthsShort = 'Jan_Feb_Mar_Apr_May_Jun_Jul_Aug_Sep_Oct_Nov_Dec'.split('_');
/** @type {?} */
const DefaultLocaleWeekdays = 'Sunday_Monday_Tuesday_Wednesday_Thursday_Friday_Saturday'.split('_');
/** @type {?} */
const DefaultLocaleWeekdaysShort = 'Sun_Mon_Tue_Wed_Thu_Fri_Sat'.split('_');

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DatepickerService {
    constructor() {
    }
    /**
     * @return {?}
     */
    get dateObj() {
        return this._dateObj;
    }
    /**
     * @param {?} value
     * @return {?}
     */
    set dateObj(value) {
        this._dateObj = value;
    }
    /**
     * @return {?}
     */
    getDatesOfMonth() {
        /** @type {?} */
        const d = this.dateObj;
        /** @type {?} */
        const y = d.getFullYear();
        /** @type {?} */
        const m = d.getMonth();
        /** @type {?} */
        const f = new Date(y, m, 1);
        /** @type {?} */
        const l = new Date(y, m + 1, 0);
        /** @type {?} */
        const days = [
            [null, null, null, null, null, null, null],
            [null, null, null, null, null, null, null],
            [null, null, null, null, null, null, null],
            [null, null, null, null, null, null, null],
            [null, null, null, null, null, null, null],
            [null, null, null, null, null, null, null]
        ];
        console.log(l.getDate());
        /** @type {?} */
        let c = f.getDay();
        /** @type {?} */
        let row = 0;
        for (let i = 1; i <= l.getDate(); i++) {
            days[row][c] = (i <= 9) ? '0' + i : i + '';
            c++;
            if (c === 7) {
                row++;
                c = 0;
            }
        }
        console.log(row, days);
        return days;
    }
    /**
     * @param {?} year
     * @param {?} month
     * @param {?} day
     * @param {?} hours
     * @param {?} minutes
     * @param {?} seconds
     * @return {?}
     */
    getDateJsonObj(year, month, day, hours, minutes, seconds) {
        this.dateObj = new Date(year, month, day, hours, minutes, seconds);
        /** @type {?} */
        const d = this.dateObj;
        /** @type {?} */
        const date = (d.getDate() <= 9) ? '0' + d.getDate() : d.getDate() + '';
        /** @type {?} */
        const monthNumeric = (d.getMonth() <= 9) ? '0' + d.getMonth() : d.getMonth() + '';
        return {
            day: d.getDay(),
            dayString: DefaultLocaleWeekdays[d.getDay()],
            dayStringShort: DefaultLocaleWeekdaysShort[d.getDay()],
            date: date,
            month: monthNumeric,
            monthString: DefaultLocaleMonths[d.getUTCMonth()],
            monthStringShort: DefaultLocaleMonthsShort[d.getUTCMonth()],
            time: d.getTime().toString(),
            hours: d.getHours().toString(),
            minutes: d.getMinutes().toString(),
            seconds: d.getSeconds().toString(),
            year: d.getFullYear().toString(),
            UTCString: d.toUTCString(),
            numericDate: d.getFullYear() + '-' + monthNumeric + '-' + date + '-' + d.getHours() + '-' + d.getSeconds(),
            formatDate: this.getFormatDate(d)
        };
    }
    /**
     * @param {?} format
     * @return {?}
     */
    setFormat(format) {
        this.format = format;
    }
    /**
     * @private
     * @param {?} d
     * @return {?}
     */
    getFormatDate(d) {
        /** @type {?} */
        let dateStr = this.format;
        console.log(d.getFullYear());
        dateStr = dateStr.replace('yyyy', d.getFullYear());
        dateStr = dateStr.replace('mm', this.getMonthString(d));
        dateStr = dateStr.replace('dd', d.getDate());
        console.log(dateStr);
        return dateStr;
    }
    /**
     * @param {?} date
     * @return {?}
     */
    getMonthString(date) {
        console.log(date.getMonth());
        /** @type {?} */
        var month = date.getMonth() + 1;
        return month < 10 ? '0' + month : '' + month; // ('' + month) for string result
    }
}
DatepickerService.ɵfac = function DatepickerService_Factory(t) { return new (t || DatepickerService)(); };
DatepickerService.ɵprov = ɵngcc0.ɵɵdefineInjectable({ token: DatepickerService, factory: DatepickerService.ɵfac, providedIn: 'root' });
DatepickerService.ctorParameters = () => [];
/** @nocollapse */ DatepickerService.ngInjectableDef = defineInjectable({ factory: function DatepickerService_Factory() { return new DatepickerService(); }, token: DatepickerService, providedIn: "root" });
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DatepickerService, [{
        type: Injectable,
        args: [{
                providedIn: 'root'
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DatepickerComponent {
    /**
     * @param {?} datepickerSvc
     */
    constructor(datepickerSvc) {
        this.datepickerSvc = datepickerSvc;
        this.selected = new EventEmitter();
        this.mode = 'single';
        this.items = [
            {
                url: 'a',
                text: 'Action',
                isDivider: true
            }, {
                url: 'a',
                text: 'Another action'
            }, {
                url: 'a',
                text: 'Something else here'
            }
        ];
        this.defaultLocaleWeekdaysShort = DefaultLocaleWeekdaysShort;
        this.range = {
            from: null,
            to: null
        };
        this.showCal = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.labelText = 'Select Date';
        /** @type {?} */
        const d = new Date();
        this.datepickerSvc.setFormat(this.format || 'yyyy-mm-dd');
        this.selectedDateTime = this.datepickerSvc.getDateJsonObj(d.getFullYear(), d.getMonth(), d.getDate(), d.getHours(), d.getMinutes(), d.getSeconds());
        this.daysList = this.datepickerSvc.getDatesOfMonth();
    }
    /**
     * @param {?} day
     * @return {?}
     */
    isWithinRange(day) {
        if (day) {
            /** @type {?} */
            const a = this.selectedDateTime.numericDate.split('-');
            a[2] = (day <= 9) ? '0' + day : day + '';
            /** @type {?} */
            const today = parseInt(a.join(''), 10);
            /** @type {?} */
            const from = parseInt(this.range.from.numericDate.split('-').join(''), 10);
            /** @type {?} */
            const to = parseInt(this.range.to.numericDate.split('-').join(''), 10);
            return (today >= from && today <= to);
        }
        return false;
    }
    /**
     * @param {?} day
     * @return {?}
     */
    getSelectedClass(day) {
        /** @type {?} */
        let isSelected = false;
        day = parseInt(day, 10);
        if (this.mode === 'range') {
            if (this.range.from) {
                isSelected = (this.range.from.date === day);
            }
            if (this.range.from && this.range.to) {
                isSelected = this.isWithinRange(day);
            }
        }
        else {
            isSelected = this.selectedDateTime.date === day;
        }
        return { 'c-datepicker__day--selected': isSelected };
    }
    /**
     * @param {?} day
     * @return {?}
     */
    setDate(day) {
        /** @type {?} */
        const d = this.selectedDateTime;
        this.selectedDateTime = this.datepickerSvc.getDateJsonObj(d.year, d.month, day, d.hours, d.minutes, d.seconds);
        if (this.mode === 'range') {
            if (this.range.from) {
                this.range.to = this.selectedDateTime;
                this.selectedDate = this.range.from.formatDate + " - " + this.range.to.formatDate;
            }
            else {
                this.range.from = this.selectedDateTime;
                this.selectedDate = this.range.from.formatDate + " ";
            }
            this.selected.emit(this.range);
        }
        else {
            this.selectedDate = this.selectedDateTime.formatDate;
            this.showCal = false;
            this.selected.emit(this.selectedDateTime);
        }
    }
    /**
     * @return {?}
     */
    prevMonth() {
        /** @type {?} */
        const d = this.selectedDateTime;
        this.selectedDateTime = this.datepickerSvc.getDateJsonObj(d.year, parseInt(d.month, 10) - 1, d.date, d.hours, d.minutes, d.seconds);
        this.daysList = this.datepickerSvc.getDatesOfMonth();
        this.selected.emit(this.selectedDateTime);
    }
    /**
     * @return {?}
     */
    nextMonth() {
        /** @type {?} */
        const d = this.selectedDateTime;
        this.selectedDateTime = this.datepickerSvc.getDateJsonObj(d.year, parseInt(d.month, 10) + 1, d.date, d.hours, d.minutes, d.seconds);
        this.daysList = this.datepickerSvc.getDatesOfMonth();
        this.selected.emit(this.selectedDateTime);
    }
    /**
     * @return {?}
     */
    clear() {
        this.range = {
            from: null,
            to: null
        };
        this.selectedDate = '';
    }
}
DatepickerComponent.ɵfac = function DatepickerComponent_Factory(t) { return new (t || DatepickerComponent)(ɵngcc0.ɵɵdirectiveInject(DatepickerService)); };
DatepickerComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DatepickerComponent, selectors: [["ui-datepicker"]], inputs: { mode: "mode", labelText: "labelText", format: "format" }, outputs: { selected: "selected" }, decls: 10, vars: 3, consts: [[1, "form-group"], ["for", "basic-url"], [1, "input-group", "mb-3"], [1, "input-group-prepend"], [1, "input-group-text"], [1, "far", "fa-calendar-alt"], ["type", "text", "placeholder", "Date", "aria-label", "Date", "aria-describedby", "basic-addon1", "id", "basic-url", 1, "form-control", 3, "ngModel", "focus"], [1, "datepicker-container"], ["class", "c-datepicker c-datepicker--open", 4, "ngIf"], [1, "c-datepicker", "c-datepicker--open"], ["title", "show time picker", 1, "c-datepicker__toggle", "c-datepicker__toggle--right", "c-datepicker--show-time", "js-show-clock"], [1, "far", "fa-clock"], [1, "c-datepicker__toggle", "c-datepicker__toggle--left", "c-datepicker--show-calendar", "js-show-calendar", "is-selected"], ["title", "clear datepicker", 1, "far", "fa-trash-alt", 3, "click"], [1, "c-datepicker__header"], [1, "c-datepicker__header-day"], [1, "js-day"], ["data-rome-id", "0", 1, "c-datepicker__calendar"], [1, "c-datepicker__calendar", 2, "display", "inline-block"], [1, "c-datepicker__date"], [1, "c-datepicker__month"], ["type", "button", 1, "c-datepicker__back", 3, "click"], [1, "fas", "fa-chevron-left"], ["type", "button", 1, "c-datepicker__next", 3, "click"], [1, "fas", "fa-chevron-right"], [1, "rd-month-label"], [1, "c-datepicker__days"], [1, "c-datepicker__days-head"], [1, "c-datepicker__days-row"], ["class", "c-datepicker__day-head", 4, "ngFor", "ngForOf"], ["data-rome-offset", "0", 1, "c-datepicker__days-body"], ["class", "c-datepicker__days-row", 4, "ngFor", "ngForOf"], [1, "c-datepicker__clock"], [1, "c-datepicker__clock__am-pm-toggle"], [1, ""], ["type", "radio", "name", "time-date-toggle", "value", "AM", 1, "c-datepicker__toggle", "c-datepicker__toggle--right", "c-datepicker__clock--am"], [1, "c-datepicker__toggle--checked"], ["type", "radio", "name", "time-date-toggle", "value", "PM", "checked", "checked", 1, "c-datepicker__toggle", "c-datepicker__toggle--right", "c-datepicker__clock--pm"], [1, "c-datepicker__mask"], [1, "c-datepicker__clock__hours", "js-clock-hours", "active"], ["data-number", "3", 1, "c-datepicker__clock__num"], ["data-number", "4", 1, "c-datepicker__clock__num"], ["data-number", "5", 1, "c-datepicker__clock__num"], ["data-number", "6", 1, "c-datepicker__clock__num"], ["data-number", "7", 1, "c-datepicker__clock__num"], ["data-number", "8", 1, "c-datepicker__clock__num"], ["data-number", "9", 1, "c-datepicker__clock__num"], ["data-number", "10", 1, "c-datepicker__clock__num"], ["data-number", "11", 1, "c-datepicker__clock__num"], ["data-number", "0", 1, "c-datepicker__clock__num", "c-datepicker__clock__num--active"], ["data-number", "1", 1, "c-datepicker__clock__num"], ["data-number", "2", 1, "c-datepicker__clock__num"], [1, "c-datepicker__clock-hands"], [1, "c-datepicker__hour-hand"], [1, "c-datepicker__clock__minutes", "js-clock-minutes"], ["data-number", "15", 1, "c-datepicker__clock__num"], ["data-number", "20", 1, "c-datepicker__clock__num"], ["data-number", "25", 1, "c-datepicker__clock__num"], ["data-number", "30", 1, "c-datepicker__clock__num"], ["data-number", "35", 1, "c-datepicker__clock__num"], ["data-number", "40", 1, "c-datepicker__clock__num"], ["data-number", "45", 1, "c-datepicker__clock__num"], ["data-number", "50", 1, "c-datepicker__clock__num"], ["data-number", "55", 1, "c-datepicker__clock__num"], [1, "modal-btns"], [1, "c-btn", "c-btn--flat", "js-cancel", 3, "click"], [1, "c-btn", "c-btn--flat", "js-ok", 3, "click"], [1, "c-datepicker__day-head"], ["class", "c-datepicker__day-body", 3, "ngClass", "click", 4, "ngFor", "ngForOf"], [1, "c-datepicker__day-body", 3, "ngClass", "click"]], template: function DatepickerComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "label", 1);
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelementStart(4, "div", 3);
        ɵngcc0.ɵɵelementStart(5, "span", 4);
        ɵngcc0.ɵɵelement(6, "i", 5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(7, "input", 6);
        ɵngcc0.ɵɵlistener("focus", function DatepickerComponent_Template_input_focus_7_listener() { return ctx.showCal = true; });
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(8, "div", 7);
        ɵngcc0.ɵɵtemplate(9, DatepickerComponent_div_9_Template, 93, 5, "div", 8);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵtextInterpolate(ctx.labelText);
        ɵngcc0.ɵɵadvance(5);
        ɵngcc0.ɵɵproperty("ngModel", ctx.selectedDate);
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showCal);
    } }, directives: [ɵngcc3.DefaultValueAccessor, ɵngcc3.NgControlStatus, ɵngcc3.NgModel, ɵngcc1.NgIf, ɵngcc1.NgForOf, ɵngcc1.NgClass], styles: [".datepicker-container[_ngcontent-%COMP%]{position:relative;z-index:999}.c-scrim[_ngcontent-%COMP%]{position:fixed;left:0;right:0;top:0;bottom:0;background-color:rgba(0,0,0,.541176);opacity:0;transition:opacity .2s;will-change:opacity}.c-scrim--shown[_ngcontent-%COMP%]{opacity:1}.c-datepicker[_ngcontent-%COMP%]{min-height:390px;position:absolute;background:#fff;border:0;width:300px;text-align:center;-webkit-tap-highlight-color:transparent;box-shadow:0 14px 45px rgba(0,0,0,.25),0 10px 18px rgba(0,0,0,.22);border-radius:2px;opacity:0;will-change:opacity;transition:opacity .2s ease-in-out,top .2s ease-in-out}.c-datepicker--open[_ngcontent-%COMP%]{opacity:1;top:50%}.c-datepicker__header[_ngcontent-%COMP%]{position:relative}.c-datepicker__header-day[_ngcontent-%COMP%]{height:32px;background:#0097a7;color:#fff;line-height:32px;font-size:12px;font-weight:200;letter-spacing:.3px}.c-datepicker__header[_ngcontent-%COMP%]::after{content:\"\";display:table;clear:both}.c-datepicker__header-date[_ngcontent-%COMP%]{background:#00bcd4;height:150px;padding:16px 0}.rd-month-label[_ngcontent-%COMP%]{height:56px;line-height:56px;font-size:14px;font-weight:800}.c-datepicker__back[_ngcontent-%COMP%], .c-datepicker__next[_ngcontent-%COMP%], .c-datepicker__toggle[_ngcontent-%COMP%]{position:absolute;border:0;background:0 0;line-height:1;width:56px;height:56px;display:inline-block;overflow:hidden;-webkit-font-smoothing:antialiased;cursor:pointer}.c-datepicker__back[_ngcontent-%COMP%]:focus, .c-datepicker__next[_ngcontent-%COMP%]:focus, .c-datepicker__toggle[_ngcontent-%COMP%]:focus{outline:0}.c-datepicker__back[_ngcontent-%COMP%]{left:0}.c-datepicker__next[_ngcontent-%COMP%]{right:0}.c-datepicker--show-calendar[_ngcontent-%COMP%]:after, .c-datepicker--show-time[_ngcontent-%COMP%]:after{color:#fff;visibility:visible}.c-datepicker__header-date[_ngcontent-%COMP%]   span[_ngcontent-%COMP%]{display:block;color:#fff;margin:0;transition:opacity .1s ease-in-out}.c-datepicker__header-date__month[_ngcontent-%COMP%]{cursor:pointer;font-size:24px;opacity:.6}.c-datepicker__header-date__day[_ngcontent-%COMP%]{cursor:pointer;font-size:64px;opacity:.6}.c-datepicker__header-date__time[_ngcontent-%COMP%]{font-size:25px;opacity:.6}.c-datepicker__header-date__time[_ngcontent-%COMP%] > span[_ngcontent-%COMP%]{display:inline-block}.c-datepicker__header-date__hours[_ngcontent-%COMP%], .c-datepicker__header-date__minutes[_ngcontent-%COMP%]{cursor:pointer}.c-datepicker--show-time.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__header[_ngcontent-%COMP%]   .c-datepicker__header-date__time[_ngcontent-%COMP%]{opacity:1}.c-datepicker--show-time.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__header[_ngcontent-%COMP%]   .c-datepicker__header-date__time[_ngcontent-%COMP%]   .c-datepicker__header-date__hours[_ngcontent-%COMP%], .c-datepicker--show-time.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__header[_ngcontent-%COMP%]   .c-datepicker__header-date__time[_ngcontent-%COMP%]   .c-datepicker__header-date__minutes[_ngcontent-%COMP%]{opacity:.6}.c-datepicker--show-calendar.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__header[_ngcontent-%COMP%]   .c-datepicker__header-date__day[_ngcontent-%COMP%], .c-datepicker--show-calendar.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__header[_ngcontent-%COMP%]   .c-datepicker__header-date__month[_ngcontent-%COMP%], .c-datepicker--show-time.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__header[_ngcontent-%COMP%]   .c-datepicker__header-date__time[_ngcontent-%COMP%]   .c-datepicker__header-date__hours.active[_ngcontent-%COMP%], .c-datepicker--show-time.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__header[_ngcontent-%COMP%]   .c-datepicker__header-date__time[_ngcontent-%COMP%]   .c-datepicker__header-date__minutes.active[_ngcontent-%COMP%]{opacity:1}.modal-btns[_ngcontent-%COMP%]{padding:20px;position:absolute;bottom:0;right:0;float:right}.c-datepicker__day-body[_ngcontent-%COMP%]{font-size:12px;color:rgba(0,0,0,.8);width:36px;height:36px}.c-datepicker__day--selected[_ngcontent-%COMP%]::after{content:\"\";position:absolute;left:50%;top:50%;width:35px;height:35px;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);background:rgba(0,0,0,.05)}.c-datepicker__day-head[_ngcontent-%COMP%]{color:rgba(0,0,0,.54);font-size:12px;height:36px}.c-datepicker__day-head[_ngcontent-%COMP%], c-datepicker__day-body[_ngcontent-%COMP%]{-webkit-tap-highlight-color:transparent}.c-btn[_ngcontent-%COMP%]{display:inline-block;min-width:56px;cursor:pointer}.rd-day-next-month[_ngcontent-%COMP%], .rd-day-prev-month[_ngcontent-%COMP%]{opacity:.1;pointer-events:none}.c-datepicker__calendar[_ngcontent-%COMP%]{height:300px}.c-datepicker__date[_ngcontent-%COMP%]{position:absolute;left:0;right:0}.c-datepicker__days[_ngcontent-%COMP%]{margin:10px 20px}.c-datepicker__header-toggle[_ngcontent-%COMP%]{position:absolute;top:50%;color:#fff;cursor:pointer}.c-datepicker__header-toggle[_ngcontent-%COMP%]   i[_ngcontent-%COMP%]{font-size:26px}.c-datepicker__header-toggle--left[_ngcontent-%COMP%]{left:20px}.c-datepicker__header-toggle--right[_ngcontent-%COMP%]{right:20px}.c-datepicker__header-toggle--inactive[_ngcontent-%COMP%]{opacity:.2}.c-datepicker__toggle[_ngcontent-%COMP%]{top:7px;width:45px;height:30px;color:#fff!important;z-index:1;transition:opacity .2s ease-in-out}.c-datepicker__toggle--right[_ngcontent-%COMP%]{right:0}.c-datepicker__toggle--left[_ngcontent-%COMP%]{left:0}.c-datepicker__toggle.is-selected[_ngcontent-%COMP%]{opacity:1}.c-datepicker--show-calendar.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__clock[_ngcontent-%COMP%], .c-datepicker--show-time.is-selected[_ngcontent-%COMP%] ~ .c-datepicker__calendar[_ngcontent-%COMP%]{display:none}.c-datepicker__clock[_ngcontent-%COMP%]{position:relative;width:200px;height:200px;padding:160px 0 20px;border-radius:50%;list-style:none;font-size:14px;line-height:50px;margin:0 auto}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]{display:block;position:absolute;top:50%;left:50%;width:50px;height:50px;margin:-25px;z-index:98}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(1){-webkit-transform:rotate(0) translate(100px) rotate(0);transform:rotate(0) translate(100px) rotate(0)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(1).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(270deg);transform:translate(-50%,-50%) rotate(270deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(1).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(1):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(270deg);transform:translate(-50%,-50%) rotate(270deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(1):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(2){-webkit-transform:rotate(30deg) translate(100px) rotate(-30deg);transform:rotate(30deg) translate(100px) rotate(-30deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(2).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(300deg);transform:translate(-50%,-50%) rotate(300deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(2).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(2):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(300deg);transform:translate(-50%,-50%) rotate(300deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(2):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(3){-webkit-transform:rotate(60deg) translate(100px) rotate(-60deg);transform:rotate(60deg) translate(100px) rotate(-60deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(3).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(330deg);transform:translate(-50%,-50%) rotate(330deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(3).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(3):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(330deg);transform:translate(-50%,-50%) rotate(330deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(3):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(4){-webkit-transform:rotate(90deg) translate(100px) rotate(-90deg);transform:rotate(90deg) translate(100px) rotate(-90deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(4).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(360deg);transform:translate(-50%,-50%) rotate(360deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(4).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(4):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(360deg);transform:translate(-50%,-50%) rotate(360deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(4):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(5){-webkit-transform:rotate(120deg) translate(100px) rotate(-120deg);transform:rotate(120deg) translate(100px) rotate(-120deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(5).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(390deg);transform:translate(-50%,-50%) rotate(390deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(5).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(5):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(390deg);transform:translate(-50%,-50%) rotate(390deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(5):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(6){-webkit-transform:rotate(150deg) translate(100px) rotate(-150deg);transform:rotate(150deg) translate(100px) rotate(-150deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(6).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(420deg);transform:translate(-50%,-50%) rotate(420deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(6).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(6):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(420deg);transform:translate(-50%,-50%) rotate(420deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(6):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(7){-webkit-transform:rotate(180deg) translate(100px) rotate(-180deg);transform:rotate(180deg) translate(100px) rotate(-180deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(7).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(450deg);transform:translate(-50%,-50%) rotate(450deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(7).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(7):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(450deg);transform:translate(-50%,-50%) rotate(450deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(7):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(8){-webkit-transform:rotate(210deg) translate(100px) rotate(-210deg);transform:rotate(210deg) translate(100px) rotate(-210deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(8).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(480deg);transform:translate(-50%,-50%) rotate(480deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(8).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(8):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(480deg);transform:translate(-50%,-50%) rotate(480deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(8):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(9){-webkit-transform:rotate(240deg) translate(100px) rotate(-240deg);transform:rotate(240deg) translate(100px) rotate(-240deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(9).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(510deg);transform:translate(-50%,-50%) rotate(510deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(9).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(9):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(510deg);transform:translate(-50%,-50%) rotate(510deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(9):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(10){-webkit-transform:rotate(270deg) translate(100px) rotate(-270deg);transform:rotate(270deg) translate(100px) rotate(-270deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(10).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(540deg);transform:translate(-50%,-50%) rotate(540deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(10).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(10):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(540deg);transform:translate(-50%,-50%) rotate(540deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(10):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(11){-webkit-transform:rotate(300deg) translate(100px) rotate(-300deg);transform:rotate(300deg) translate(100px) rotate(-300deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(11).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(570deg);transform:translate(-50%,-50%) rotate(570deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(11).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(11):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(570deg);transform:translate(-50%,-50%) rotate(570deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(11):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(12){-webkit-transform:rotate(330deg) translate(100px) rotate(-330deg);transform:rotate(330deg) translate(100px) rotate(-330deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(12).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(600deg);transform:translate(-50%,-50%) rotate(600deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(12).c-datepicker__clock__num--active:not(.hide-hand) ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(12):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]{-webkit-transform:translate(-50%,-50%) rotate(600deg);transform:translate(-50%,-50%) rotate(600deg)}.c-datepicker__clock[_ngcontent-%COMP%]   .c-datepicker__clock__num[_ngcontent-%COMP%]:nth-of-type(12):hover ~ .c-datepicker__clock-hands[_ngcontent-%COMP%]   .c-datepicker__hour-hand[_ngcontent-%COMP%]{opacity:1;background:#00bcd4}.c-datepicker__clock[_ngcontent-%COMP%]::before{content:\"\";position:absolute;top:70px;left:-20px;width:240px;height:240px;background:rgba(0,0,0,.05);border-radius:50%}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]   label[_ngcontent-%COMP%], .c-datepicker__clock__num[_ngcontent-%COMP%], .c-datepicker__day-body[_ngcontent-%COMP%], .u-hover-ball-effect[_ngcontent-%COMP%]{position:relative;cursor:pointer}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]   label[_ngcontent-%COMP%]:before, .c-datepicker__clock__num[_ngcontent-%COMP%]:before, .c-datepicker__day-body[_ngcontent-%COMP%]:before, .u-hover-ball-effect[_ngcontent-%COMP%]:before{content:\"\";position:absolute;left:50%;top:50%;width:0%;height:0%;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);transition:width .1s ease-in-out,height .1s ease-in-out}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]   label[_ngcontent-%COMP%]:hover, .c-datepicker__clock__num[_ngcontent-%COMP%]:hover, .c-datepicker__day-body[_ngcontent-%COMP%]:hover, .u-hover-ball-effect[_ngcontent-%COMP%]:hover{color:#fff}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]   label[_ngcontent-%COMP%]:hover:before, .c-datepicker__clock__num[_ngcontent-%COMP%]:hover:before, .c-datepicker__day-body[_ngcontent-%COMP%]:hover:before, .u-hover-ball-effect[_ngcontent-%COMP%]:hover:before{background:#00bcd4;width:35px;height:35px;z-index:-1}.c-datepicker__clock__num--active[_ngcontent-%COMP%]:not(.hide-hand), .c-datepicker__day-body--active[_ngcontent-%COMP%]:not(.hide-hand){color:#fff}.c-datepicker__clock__num--active[_ngcontent-%COMP%]:not(.hide-hand):before, .c-datepicker__day-body--active[_ngcontent-%COMP%]:not(.hide-hand):before{background:#00bcd4;width:35px;height:35px;z-index:-1}.c-datepicker__clock-hands[_ngcontent-%COMP%]{position:absolute;left:50%;top:50%;-webkit-transform:translate(-50%,-50%) rotate(180deg);transform:translate(-50%,-50%) rotate(180deg);width:10px;height:10px;border-radius:50%;background:#0097a7}.c-datepicker__hour-hand[_ngcontent-%COMP%]{position:absolute;opacity:0;height:78px;width:2px;background:#00bcd4;left:4px;top:10px}.c-datepicker__clock__minutes[_ngcontent-%COMP%]{height:200px;margin:-69px 0 0;width:200px;display:none}.c-datepicker__clock__minutes.active[_ngcontent-%COMP%]{display:block}.c-datepicker__clock__hours[_ngcontent-%COMP%]{height:200px;margin:-69px 0 0;width:200px;display:none}.c-datepicker__clock__hours.active[_ngcontent-%COMP%]{display:block}.c-datepicker__mask[_ngcontent-%COMP%]{width:127px;height:132px;position:absolute;top:122px;left:37px;z-index:99}.c-datepicker__mask[_ngcontent-%COMP%]:after{content:' ';width:156px;height:70px;display:block;position:absolute;top:32px;left:0;margin-left:-13px}.c-datepicker__mask[_ngcontent-%COMP%]:before{content:' ';width:75px;height:158px;display:block;position:absolute;top:6px;left:28px;margin-top:-18px}.c-datepicker__clock--show-minutes[_ngcontent-%COMP%]   .c-datepicker__clock__minutes[_ngcontent-%COMP%]{visibility:visible}.c-datepicker__clock--show-hours[_ngcontent-%COMP%]   .c-datepicker__clock__minutes[_ngcontent-%COMP%], .c-datepicker__clock--show-minutes[_ngcontent-%COMP%]   .c-datepicker__clock__hours[_ngcontent-%COMP%]{visibility:hidden}.c-datepicker__clock--show-hours[_ngcontent-%COMP%]   .c-datepicker__clock__hours[_ngcontent-%COMP%]{visibility:visible}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]{position:absolute;top:0;left:10px;right:10px;height:40px;padding:20px;line-height:40px}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]   label[_ngcontent-%COMP%]{width:40px;position:absolute}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]   label[_ngcontent-%COMP%]:nth-child(1){left:0}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]   label[_ngcontent-%COMP%]:nth-child(2){right:0}.c-datepicker__clock__am-pm-toggle[_ngcontent-%COMP%]   label.c-datepicker__toggle--checked[_ngcontent-%COMP%]::after{content:\"\";position:absolute;left:50%;top:50%;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);width:36px;height:36px;z-index:-1;background:rgba(0,0,0,.05)}.input-group[_ngcontent-%COMP%]{position:relative;display:flex;flex-wrap:wrap;align-items:stretch;width:100%}.input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .form-control-plaintext[_ngcontent-%COMP%]{position:relative;flex:1 1 auto;width:1%;margin-bottom:0}.input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%] + .custom-file[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%] + .custom-select[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%] + .form-control[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%] + .custom-file[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%] + .custom-select[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%] + .form-control[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%] + .custom-file[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%] + .custom-select[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%] + .form-control[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .form-control-plaintext[_ngcontent-%COMP%] + .custom-file[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .form-control-plaintext[_ngcontent-%COMP%] + .custom-select[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .form-control-plaintext[_ngcontent-%COMP%] + .form-control[_ngcontent-%COMP%]{margin-left:-1px}.input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%]   .custom-file-input[_ngcontent-%COMP%]:focus ~ .custom-file-label[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%]:focus, .input-group[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%]:focus{z-index:3}.input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%]   .custom-file-input[_ngcontent-%COMP%]:focus{z-index:4}.input-group[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%]:not(:last-child), .input-group[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%]:not(:last-child){border-top-right-radius:0;border-bottom-right-radius:0}.input-group[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%]:not(:first-child), .input-group[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%]:not(:first-child){border-top-left-radius:0;border-bottom-left-radius:0}.input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%]{display:flex;align-items:center}.input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%]:not(:last-child)   .custom-file-label[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%]:not(:last-child)   .custom-file-label[_ngcontent-%COMP%]::after{border-top-right-radius:0;border-bottom-right-radius:0}.input-group[_ngcontent-%COMP%] > .custom-file[_ngcontent-%COMP%]:not(:first-child)   .custom-file-label[_ngcontent-%COMP%]{border-top-left-radius:0;border-bottom-left-radius:0}.input-group-append[_ngcontent-%COMP%], .input-group-prepend[_ngcontent-%COMP%]{display:flex}.input-group-append[_ngcontent-%COMP%]   .btn[_ngcontent-%COMP%], .input-group-prepend[_ngcontent-%COMP%]   .btn[_ngcontent-%COMP%]{position:relative;z-index:2}.input-group-append[_ngcontent-%COMP%]   .btn[_ngcontent-%COMP%]:focus, .input-group-prepend[_ngcontent-%COMP%]   .btn[_ngcontent-%COMP%]:focus{z-index:3}.input-group-append[_ngcontent-%COMP%]   .btn[_ngcontent-%COMP%] + .btn[_ngcontent-%COMP%], .input-group-append[_ngcontent-%COMP%]   .btn[_ngcontent-%COMP%] + .input-group-text[_ngcontent-%COMP%], .input-group-append[_ngcontent-%COMP%]   .input-group-text[_ngcontent-%COMP%] + .btn[_ngcontent-%COMP%], .input-group-append[_ngcontent-%COMP%]   .input-group-text[_ngcontent-%COMP%] + .input-group-text[_ngcontent-%COMP%], .input-group-prepend[_ngcontent-%COMP%]   .btn[_ngcontent-%COMP%] + .btn[_ngcontent-%COMP%], .input-group-prepend[_ngcontent-%COMP%]   .btn[_ngcontent-%COMP%] + .input-group-text[_ngcontent-%COMP%], .input-group-prepend[_ngcontent-%COMP%]   .input-group-text[_ngcontent-%COMP%] + .btn[_ngcontent-%COMP%], .input-group-prepend[_ngcontent-%COMP%]   .input-group-text[_ngcontent-%COMP%] + .input-group-text[_ngcontent-%COMP%]{margin-left:-1px}.input-group-prepend[_ngcontent-%COMP%]{margin-right:-1px}.input-group-append[_ngcontent-%COMP%]{margin-left:-1px}.input-group-text[_ngcontent-%COMP%]{display:flex;align-items:center;padding:.375rem .75rem;margin-bottom:0;font-size:1rem;font-weight:400;line-height:1.5;color:#495057;text-align:center;white-space:nowrap;background-color:#e9ecef;border:1px solid #ced4da;border-radius:.25rem}.input-group-text[_ngcontent-%COMP%]   input[type=checkbox][_ngcontent-%COMP%], .input-group-text[_ngcontent-%COMP%]   input[type=radio][_ngcontent-%COMP%]{margin-top:0}.input-group-lg[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%], .input-group-lg[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%]:not(textarea){height:calc(2.875rem + 2px)}.input-group-lg[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%], .input-group-lg[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%], .input-group-lg[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%] > .btn[_ngcontent-%COMP%], .input-group-lg[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%] > .input-group-text[_ngcontent-%COMP%], .input-group-lg[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%] > .btn[_ngcontent-%COMP%], .input-group-lg[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%] > .input-group-text[_ngcontent-%COMP%]{padding:.5rem 1rem;font-size:1.25rem;line-height:1.5;border-radius:.3rem}.input-group-sm[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%], .input-group-sm[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%]:not(textarea){height:calc(1.8125rem + 2px)}.input-group-sm[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%], .input-group-sm[_ngcontent-%COMP%] > .form-control[_ngcontent-%COMP%], .input-group-sm[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%] > .btn[_ngcontent-%COMP%], .input-group-sm[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%] > .input-group-text[_ngcontent-%COMP%], .input-group-sm[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%] > .btn[_ngcontent-%COMP%], .input-group-sm[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%] > .input-group-text[_ngcontent-%COMP%]{padding:.25rem .5rem;font-size:.875rem;line-height:1.5;border-radius:.2rem}.input-group-lg[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%], .input-group-sm[_ngcontent-%COMP%] > .custom-select[_ngcontent-%COMP%]{padding-right:1.75rem}.input-group[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%]:last-child > .btn[_ngcontent-%COMP%]:not(:last-child):not(.dropdown-toggle), .input-group[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%]:last-child > .input-group-text[_ngcontent-%COMP%]:not(:last-child), .input-group[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%]:not(:last-child) > .btn[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%]:not(:last-child) > .input-group-text[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%] > .btn[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%] > .input-group-text[_ngcontent-%COMP%]{border-top-right-radius:0;border-bottom-right-radius:0}.input-group[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%] > .btn[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .input-group-append[_ngcontent-%COMP%] > .input-group-text[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%]:first-child > .btn[_ngcontent-%COMP%]:not(:first-child), .input-group[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%]:first-child > .input-group-text[_ngcontent-%COMP%]:not(:first-child), .input-group[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%]:not(:first-child) > .btn[_ngcontent-%COMP%], .input-group[_ngcontent-%COMP%] > .input-group-prepend[_ngcontent-%COMP%]:not(:first-child) > .input-group-text[_ngcontent-%COMP%]{border-top-left-radius:0;border-bottom-left-radius:0}"] });
DatepickerComponent.ctorParameters = () => [
    { type: DatepickerService }
];
DatepickerComponent.propDecorators = {
    selected: [{ type: Output }],
    mode: [{ type: Input }],
    format: [{ type: Input }],
    labelText: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DatepickerComponent, [{
        type: Component,
        args: [{
                selector: 'ui-datepicker',
                template: `<div class="form-group">
  <label for="basic-url">{{labelText}}</label>
  <div class="input-group mb-3">
    <div class="input-group-prepend">
      <span class="input-group-text">
        <i class="far fa-calendar-alt"></i>
      </span>
    </div>
    <input type="text" class="form-control" placeholder="Date" (focus)="showCal=true" aria-label="Date"
           aria-describedby="basic-addon1" [ngModel]="selectedDate" id="basic-url">
  </div>
</div>
<div class="datepicker-container">
  <div class="c-datepicker c-datepicker--open" *ngIf="showCal">
    <a class="c-datepicker__toggle c-datepicker__toggle--right c-datepicker--show-time js-show-clock"
       title="show time picker">
      <i class="far fa-clock"></i>
    </a>

    <div
      class="c-datepicker__toggle c-datepicker__toggle--left c-datepicker--show-calendar js-show-calendar is-selected">
      <!--&nbsp;
      <a class=""
         title="show date picker">
        <i class="far fa-calendar-alt"></i>
      </a>
      &nbsp;-->

      <i class="far fa-trash-alt" title="clear datepicker" (click)="clear()"></i>

    </div>


    <div class="c-datepicker__header">
      <div class="c-datepicker__header-day">
        <span class="js-day">{{selectedDateTime.formatDate}}</span>
      </div>
      <!--<div class="c-datepicker__header-date">-->
      <!--<span class="c-datepicker__header-date__month js-date-month">Jun 2019</span>-->
      <!--<span class="c-datepicker__header-date__day js-date-day">10th</span>-->
      <!--<span class="c-datepicker__header-date__time js-date-time">-->
      <!--<span class="c-datepicker__header-date__hours js-date-hours active">12</span>:<span-->
      <!--class="c-datepicker__header-date__minutes js-date-minutes">00</span>-->
      <!--</span>-->
      <!--</div>-->
    </div>

    <div class="c-datepicker__calendar" data-rome-id="0">
      <div class="c-datepicker__calendar" style="display: inline-block;">
        <div class="c-datepicker__date">
          <div class="c-datepicker__month">
            <button class="c-datepicker__back" type="button" (click)="prevMonth()"><i class="fas fa-chevron-left"></i>
            </button>
            <button class="c-datepicker__next" type="button" (click)="nextMonth()"><i class="fas fa-chevron-right"></i>
            </button>
            <div class="rd-month-label">{{selectedDateTime.monthStringShort}} {{selectedDateTime.year}}</div>
            <table class="c-datepicker__days">
              <thead class="c-datepicker__days-head">
              <!-- Days Sun - Sat -->
              <tr class="c-datepicker__days-row">
                <th class="c-datepicker__day-head" *ngFor="let day  of defaultLocaleWeekdaysShort">{{day}}</th>
              </tr>
              </thead>
              <tbody class="c-datepicker__days-body" data-rome-offset="0">
              <!-- Date 1-31 -->
              <tr class="c-datepicker__days-row" *ngFor="let row of daysList">
                <td class="c-datepicker__day-body" *ngFor="let day of row" [ngClass]="getSelectedClass(day)"
                    (click)="setDate(day)">
                  <span>{{day}}</span>
                </td>
              </tr>
              <!--<tr class="c-datepicker__days-row">-->
              <!--<td class="c-datepicker__day-body rd-day-prev-month">26</td>-->
              <!--<td class="c-datepicker__day-body c-datepicker__day selected">10</td>-->
              <!--<td class="c-datepicker__day-body rd-day-next-month">01</td>-->
              <!--</tr>  -->
              </tbody>
            </table>
          </div>
        </div>
      </div>
    </div>
    <div class="c-datepicker__clock">
      <div class="c-datepicker__clock__am-pm-toggle">
        <label class="">
          <input class="c-datepicker__toggle c-datepicker__toggle--right c-datepicker__clock--am" type="radio"
                 name="time-date-toggle" value="AM">
          AM
        </label>

        <label class="c-datepicker__toggle--checked">
          <input class="c-datepicker__toggle c-datepicker__toggle--right c-datepicker__clock--pm" type="radio"
                 name="time-date-toggle" value="PM" checked="checked">
          PM
        </label>
      </div>
      <div class="c-datepicker__mask"></div>
      <div class="c-datepicker__clock__hours js-clock-hours active">
        <div class="c-datepicker__clock__num" data-number="3">3</div>
        <div class="c-datepicker__clock__num" data-number="4">4</div>
        <div class="c-datepicker__clock__num" data-number="5">5</div>
        <div class="c-datepicker__clock__num" data-number="6">6</div>
        <div class="c-datepicker__clock__num" data-number="7">7</div>
        <div class="c-datepicker__clock__num" data-number="8">8</div>
        <div class="c-datepicker__clock__num" data-number="9">9</div>
        <div class="c-datepicker__clock__num" data-number="10">10</div>
        <div class="c-datepicker__clock__num" data-number="11">11</div>
        <div class="c-datepicker__clock__num c-datepicker__clock__num--active" data-number="0">12</div>
        <div class="c-datepicker__clock__num" data-number="1">1</div>
        <div class="c-datepicker__clock__num" data-number="2">2</div>
        <div class="c-datepicker__clock-hands">
          <div class="c-datepicker__hour-hand"></div>
        </div>
      </div>
      <div class="c-datepicker__clock__minutes js-clock-minutes">
        <div class="c-datepicker__clock__num" data-number="15">15</div>
        <div class="c-datepicker__clock__num" data-number="20">20</div>
        <div class="c-datepicker__clock__num" data-number="25">25</div>
        <div class="c-datepicker__clock__num" data-number="30">30</div>
        <div class="c-datepicker__clock__num" data-number="35">35</div>
        <div class="c-datepicker__clock__num" data-number="40">40</div>
        <div class="c-datepicker__clock__num" data-number="45">45</div>
        <div class="c-datepicker__clock__num" data-number="50">50</div>
        <div class="c-datepicker__clock__num" data-number="55">55</div>
        <div class="c-datepicker__clock__num c-datepicker__clock__num--active" data-number="0">0</div>
        <div class="c-datepicker__clock__num" data-number="5">5</div>
        <div class="c-datepicker__clock__num" data-number="10">10</div>
        <div class="c-datepicker__clock-hands">
          <div class="c-datepicker__hour-hand"></div>
        </div>
      </div>
    </div>
    <div class="modal-btns">
      <a class="c-btn c-btn--flat js-cancel" (click)="clear()">Cancel</a>
      <a class="c-btn c-btn--flat js-ok" (click)="showCal=false">OK</a>
    </div>
  </div>
</div>
`,
                styles: [`.datepicker-container{position:relative;z-index:999}.c-scrim{position:fixed;left:0;right:0;top:0;bottom:0;background-color:rgba(0,0,0,.541176);opacity:0;transition:opacity .2s;will-change:opacity}.c-scrim--shown{opacity:1}.c-datepicker{min-height:390px;position:absolute;background:#fff;border:0;width:300px;text-align:center;-webkit-tap-highlight-color:transparent;box-shadow:0 14px 45px rgba(0,0,0,.25),0 10px 18px rgba(0,0,0,.22);border-radius:2px;opacity:0;will-change:opacity;transition:opacity .2s ease-in-out,top .2s ease-in-out}.c-datepicker--open{opacity:1;top:50%}.c-datepicker__header{position:relative}.c-datepicker__header-day{height:32px;background:#0097a7;color:#fff;line-height:32px;font-size:12px;font-weight:200;letter-spacing:.3px}.c-datepicker__header::after{content:"";display:table;clear:both}.c-datepicker__header-date{background:#00bcd4;height:150px;padding:16px 0}.rd-month-label{height:56px;line-height:56px;font-size:14px;font-weight:800}.c-datepicker__back,.c-datepicker__next,.c-datepicker__toggle{position:absolute;border:0;background:0 0;line-height:1;width:56px;height:56px;display:inline-block;overflow:hidden;-webkit-font-smoothing:antialiased;cursor:pointer}.c-datepicker__back:focus,.c-datepicker__next:focus,.c-datepicker__toggle:focus{outline:0}.c-datepicker__back{left:0}.c-datepicker__next{right:0}.c-datepicker--show-calendar:after,.c-datepicker--show-time:after{color:#fff;visibility:visible}.c-datepicker__header-date span{display:block;color:#fff;margin:0;transition:opacity .1s ease-in-out}.c-datepicker__header-date__month{cursor:pointer;font-size:24px;opacity:.6}.c-datepicker__header-date__day{cursor:pointer;font-size:64px;opacity:.6}.c-datepicker__header-date__time{font-size:25px;opacity:.6}.c-datepicker__header-date__time>span{display:inline-block}.c-datepicker__header-date__hours,.c-datepicker__header-date__minutes{cursor:pointer}.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time{opacity:1}.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time .c-datepicker__header-date__hours,.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time .c-datepicker__header-date__minutes{opacity:.6}.c-datepicker--show-calendar.is-selected~.c-datepicker__header .c-datepicker__header-date__day,.c-datepicker--show-calendar.is-selected~.c-datepicker__header .c-datepicker__header-date__month,.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time .c-datepicker__header-date__hours.active,.c-datepicker--show-time.is-selected~.c-datepicker__header .c-datepicker__header-date__time .c-datepicker__header-date__minutes.active{opacity:1}.modal-btns{padding:20px;position:absolute;bottom:0;right:0;float:right}.c-datepicker__day-body{font-size:12px;color:rgba(0,0,0,.8);width:36px;height:36px}.c-datepicker__day--selected::after{content:"";position:absolute;left:50%;top:50%;width:35px;height:35px;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);background:rgba(0,0,0,.05)}.c-datepicker__day-head{color:rgba(0,0,0,.54);font-size:12px;height:36px}.c-datepicker__day-head,c-datepicker__day-body{-webkit-tap-highlight-color:transparent}.c-btn{display:inline-block;min-width:56px;cursor:pointer}.rd-day-next-month,.rd-day-prev-month{opacity:.1;pointer-events:none}.c-datepicker__calendar{height:300px}.c-datepicker__date{position:absolute;left:0;right:0}.c-datepicker__days{margin:10px 20px}.c-datepicker__header-toggle{position:absolute;top:50%;color:#fff;cursor:pointer}.c-datepicker__header-toggle i{font-size:26px}.c-datepicker__header-toggle--left{left:20px}.c-datepicker__header-toggle--right{right:20px}.c-datepicker__header-toggle--inactive{opacity:.2}.c-datepicker__toggle{top:7px;width:45px;height:30px;color:#fff!important;z-index:1;transition:opacity .2s ease-in-out}.c-datepicker__toggle--right{right:0}.c-datepicker__toggle--left{left:0}.c-datepicker__toggle.is-selected{opacity:1}.c-datepicker--show-calendar.is-selected~.c-datepicker__clock,.c-datepicker--show-time.is-selected~.c-datepicker__calendar{display:none}.c-datepicker__clock{position:relative;width:200px;height:200px;padding:160px 0 20px;border-radius:50%;list-style:none;font-size:14px;line-height:50px;margin:0 auto}.c-datepicker__clock .c-datepicker__clock__num{display:block;position:absolute;top:50%;left:50%;width:50px;height:50px;margin:-25px;z-index:98}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1){-webkit-transform:rotate(0) translate(100px) rotate(0);transform:rotate(0) translate(100px) rotate(0)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(270deg);transform:translate(-50%,-50%) rotate(270deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(270deg);transform:translate(-50%,-50%) rotate(270deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(1):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2){-webkit-transform:rotate(30deg) translate(100px) rotate(-30deg);transform:rotate(30deg) translate(100px) rotate(-30deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(300deg);transform:translate(-50%,-50%) rotate(300deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(300deg);transform:translate(-50%,-50%) rotate(300deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(2):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3){-webkit-transform:rotate(60deg) translate(100px) rotate(-60deg);transform:rotate(60deg) translate(100px) rotate(-60deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(330deg);transform:translate(-50%,-50%) rotate(330deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(330deg);transform:translate(-50%,-50%) rotate(330deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(3):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4){-webkit-transform:rotate(90deg) translate(100px) rotate(-90deg);transform:rotate(90deg) translate(100px) rotate(-90deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(360deg);transform:translate(-50%,-50%) rotate(360deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(360deg);transform:translate(-50%,-50%) rotate(360deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(4):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5){-webkit-transform:rotate(120deg) translate(100px) rotate(-120deg);transform:rotate(120deg) translate(100px) rotate(-120deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(390deg);transform:translate(-50%,-50%) rotate(390deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(390deg);transform:translate(-50%,-50%) rotate(390deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(5):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6){-webkit-transform:rotate(150deg) translate(100px) rotate(-150deg);transform:rotate(150deg) translate(100px) rotate(-150deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(420deg);transform:translate(-50%,-50%) rotate(420deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(420deg);transform:translate(-50%,-50%) rotate(420deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(6):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7){-webkit-transform:rotate(180deg) translate(100px) rotate(-180deg);transform:rotate(180deg) translate(100px) rotate(-180deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(450deg);transform:translate(-50%,-50%) rotate(450deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(450deg);transform:translate(-50%,-50%) rotate(450deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(7):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8){-webkit-transform:rotate(210deg) translate(100px) rotate(-210deg);transform:rotate(210deg) translate(100px) rotate(-210deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(480deg);transform:translate(-50%,-50%) rotate(480deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(480deg);transform:translate(-50%,-50%) rotate(480deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(8):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9){-webkit-transform:rotate(240deg) translate(100px) rotate(-240deg);transform:rotate(240deg) translate(100px) rotate(-240deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(510deg);transform:translate(-50%,-50%) rotate(510deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(510deg);transform:translate(-50%,-50%) rotate(510deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(9):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10){-webkit-transform:rotate(270deg) translate(100px) rotate(-270deg);transform:rotate(270deg) translate(100px) rotate(-270deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(540deg);transform:translate(-50%,-50%) rotate(540deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(540deg);transform:translate(-50%,-50%) rotate(540deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(10):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11){-webkit-transform:rotate(300deg) translate(100px) rotate(-300deg);transform:rotate(300deg) translate(100px) rotate(-300deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(570deg);transform:translate(-50%,-50%) rotate(570deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(570deg);transform:translate(-50%,-50%) rotate(570deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(11):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12){-webkit-transform:rotate(330deg) translate(100px) rotate(-330deg);transform:rotate(330deg) translate(100px) rotate(-330deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(600deg);transform:translate(-50%,-50%) rotate(600deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12).c-datepicker__clock__num--active:not(.hide-hand)~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12):hover~.c-datepicker__clock-hands{-webkit-transform:translate(-50%,-50%) rotate(600deg);transform:translate(-50%,-50%) rotate(600deg)}.c-datepicker__clock .c-datepicker__clock__num:nth-of-type(12):hover~.c-datepicker__clock-hands .c-datepicker__hour-hand{opacity:1;background:#00bcd4}.c-datepicker__clock::before{content:"";position:absolute;top:70px;left:-20px;width:240px;height:240px;background:rgba(0,0,0,.05);border-radius:50%}.c-datepicker__clock__am-pm-toggle label,.c-datepicker__clock__num,.c-datepicker__day-body,.u-hover-ball-effect{position:relative;cursor:pointer}.c-datepicker__clock__am-pm-toggle label:before,.c-datepicker__clock__num:before,.c-datepicker__day-body:before,.u-hover-ball-effect:before{content:"";position:absolute;left:50%;top:50%;width:0%;height:0%;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);transition:width .1s ease-in-out,height .1s ease-in-out}.c-datepicker__clock__am-pm-toggle label:hover,.c-datepicker__clock__num:hover,.c-datepicker__day-body:hover,.u-hover-ball-effect:hover{color:#fff}.c-datepicker__clock__am-pm-toggle label:hover:before,.c-datepicker__clock__num:hover:before,.c-datepicker__day-body:hover:before,.u-hover-ball-effect:hover:before{background:#00bcd4;width:35px;height:35px;z-index:-1}.c-datepicker__clock__num--active:not(.hide-hand),.c-datepicker__day-body--active:not(.hide-hand){color:#fff}.c-datepicker__clock__num--active:not(.hide-hand):before,.c-datepicker__day-body--active:not(.hide-hand):before{background:#00bcd4;width:35px;height:35px;z-index:-1}.c-datepicker__clock-hands{position:absolute;left:50%;top:50%;-webkit-transform:translate(-50%,-50%) rotate(180deg);transform:translate(-50%,-50%) rotate(180deg);width:10px;height:10px;border-radius:50%;background:#0097a7}.c-datepicker__hour-hand{position:absolute;opacity:0;height:78px;width:2px;background:#00bcd4;left:4px;top:10px}.c-datepicker__clock__minutes{height:200px;margin:-69px 0 0;width:200px;display:none}.c-datepicker__clock__minutes.active{display:block}.c-datepicker__clock__hours{height:200px;margin:-69px 0 0;width:200px;display:none}.c-datepicker__clock__hours.active{display:block}.c-datepicker__mask{width:127px;height:132px;position:absolute;top:122px;left:37px;z-index:99}.c-datepicker__mask:after{content:' ';width:156px;height:70px;display:block;position:absolute;top:32px;left:0;margin-left:-13px}.c-datepicker__mask:before{content:' ';width:75px;height:158px;display:block;position:absolute;top:6px;left:28px;margin-top:-18px}.c-datepicker__clock--show-minutes .c-datepicker__clock__minutes{visibility:visible}.c-datepicker__clock--show-hours .c-datepicker__clock__minutes,.c-datepicker__clock--show-minutes .c-datepicker__clock__hours{visibility:hidden}.c-datepicker__clock--show-hours .c-datepicker__clock__hours{visibility:visible}.c-datepicker__clock__am-pm-toggle{position:absolute;top:0;left:10px;right:10px;height:40px;padding:20px;line-height:40px}.c-datepicker__clock__am-pm-toggle label{width:40px;position:absolute}.c-datepicker__clock__am-pm-toggle label:nth-child(1){left:0}.c-datepicker__clock__am-pm-toggle label:nth-child(2){right:0}.c-datepicker__clock__am-pm-toggle label.c-datepicker__toggle--checked::after{content:"";position:absolute;left:50%;top:50%;border-radius:50%;-webkit-transform:translate(-50%,-50%);transform:translate(-50%,-50%);width:36px;height:36px;z-index:-1;background:rgba(0,0,0,.05)}.input-group{position:relative;display:flex;flex-wrap:wrap;align-items:stretch;width:100%}.input-group>.custom-file,.input-group>.custom-select,.input-group>.form-control,.input-group>.form-control-plaintext{position:relative;flex:1 1 auto;width:1%;margin-bottom:0}.input-group>.custom-file+.custom-file,.input-group>.custom-file+.custom-select,.input-group>.custom-file+.form-control,.input-group>.custom-select+.custom-file,.input-group>.custom-select+.custom-select,.input-group>.custom-select+.form-control,.input-group>.form-control+.custom-file,.input-group>.form-control+.custom-select,.input-group>.form-control+.form-control,.input-group>.form-control-plaintext+.custom-file,.input-group>.form-control-plaintext+.custom-select,.input-group>.form-control-plaintext+.form-control{margin-left:-1px}.input-group>.custom-file .custom-file-input:focus~.custom-file-label,.input-group>.custom-select:focus,.input-group>.form-control:focus{z-index:3}.input-group>.custom-file .custom-file-input:focus{z-index:4}.input-group>.custom-select:not(:last-child),.input-group>.form-control:not(:last-child){border-top-right-radius:0;border-bottom-right-radius:0}.input-group>.custom-select:not(:first-child),.input-group>.form-control:not(:first-child){border-top-left-radius:0;border-bottom-left-radius:0}.input-group>.custom-file{display:flex;align-items:center}.input-group>.custom-file:not(:last-child) .custom-file-label,.input-group>.custom-file:not(:last-child) .custom-file-label::after{border-top-right-radius:0;border-bottom-right-radius:0}.input-group>.custom-file:not(:first-child) .custom-file-label{border-top-left-radius:0;border-bottom-left-radius:0}.input-group-append,.input-group-prepend{display:flex}.input-group-append .btn,.input-group-prepend .btn{position:relative;z-index:2}.input-group-append .btn:focus,.input-group-prepend .btn:focus{z-index:3}.input-group-append .btn+.btn,.input-group-append .btn+.input-group-text,.input-group-append .input-group-text+.btn,.input-group-append .input-group-text+.input-group-text,.input-group-prepend .btn+.btn,.input-group-prepend .btn+.input-group-text,.input-group-prepend .input-group-text+.btn,.input-group-prepend .input-group-text+.input-group-text{margin-left:-1px}.input-group-prepend{margin-right:-1px}.input-group-append{margin-left:-1px}.input-group-text{display:flex;align-items:center;padding:.375rem .75rem;margin-bottom:0;font-size:1rem;font-weight:400;line-height:1.5;color:#495057;text-align:center;white-space:nowrap;background-color:#e9ecef;border:1px solid #ced4da;border-radius:.25rem}.input-group-text input[type=checkbox],.input-group-text input[type=radio]{margin-top:0}.input-group-lg>.custom-select,.input-group-lg>.form-control:not(textarea){height:calc(2.875rem + 2px)}.input-group-lg>.custom-select,.input-group-lg>.form-control,.input-group-lg>.input-group-append>.btn,.input-group-lg>.input-group-append>.input-group-text,.input-group-lg>.input-group-prepend>.btn,.input-group-lg>.input-group-prepend>.input-group-text{padding:.5rem 1rem;font-size:1.25rem;line-height:1.5;border-radius:.3rem}.input-group-sm>.custom-select,.input-group-sm>.form-control:not(textarea){height:calc(1.8125rem + 2px)}.input-group-sm>.custom-select,.input-group-sm>.form-control,.input-group-sm>.input-group-append>.btn,.input-group-sm>.input-group-append>.input-group-text,.input-group-sm>.input-group-prepend>.btn,.input-group-sm>.input-group-prepend>.input-group-text{padding:.25rem .5rem;font-size:.875rem;line-height:1.5;border-radius:.2rem}.input-group-lg>.custom-select,.input-group-sm>.custom-select{padding-right:1.75rem}.input-group>.input-group-append:last-child>.btn:not(:last-child):not(.dropdown-toggle),.input-group>.input-group-append:last-child>.input-group-text:not(:last-child),.input-group>.input-group-append:not(:last-child)>.btn,.input-group>.input-group-append:not(:last-child)>.input-group-text,.input-group>.input-group-prepend>.btn,.input-group>.input-group-prepend>.input-group-text{border-top-right-radius:0;border-bottom-right-radius:0}.input-group>.input-group-append>.btn,.input-group>.input-group-append>.input-group-text,.input-group>.input-group-prepend:first-child>.btn:not(:first-child),.input-group>.input-group-prepend:first-child>.input-group-text:not(:first-child),.input-group>.input-group-prepend:not(:first-child)>.btn,.input-group>.input-group-prepend:not(:first-child)>.input-group-text{border-top-left-radius:0;border-bottom-left-radius:0}`]
            }]
    }], function () { return [{ type: DatepickerService }]; }, { selected: [{
            type: Output
        }], mode: [{
            type: Input
        }], labelText: [{
            type: Input
        }], format: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DropdownComponent {
    constructor() {
        this.items = [];
        this.buttonText = 'Your Button Text';
        this.isShowDropdown = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.isShowDropdown = this.isOpen || false;
    }
    /**
     * @return {?}
     */
    toggle() {
        this.isShowDropdown = !this.isShowDropdown;
    }
}
DropdownComponent.ɵfac = function DropdownComponent_Factory(t) { return new (t || DropdownComponent)(); };
DropdownComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: DropdownComponent, selectors: [["ui-dropdown"]], inputs: { items: "items", buttonText: "buttonText", isOpen: "isOpen" }, ngContentSelectors: _c0, decls: 6, vars: 9, consts: [[1, "dropdown", "show", 3, "ngClass"], ["role", "button", "id", "dropdownMenuLink", "data-toggle", "dropdown", "aria-haspopup", "true", 1, "btn", "btn-secondary", "dropdown-toggle", 3, "click"], ["aria-labelledby", "dropdownMenuLink", 1, "dropdown-menu", "dropdown-menu-right", 3, "ngClass"], [4, "ngFor", "ngForOf"], [1, "dropdown-item", 3, "href"], ["class", "dropdown-divider", 4, "ngIf"], [1, "dropdown-divider"]], template: function DropdownComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵelementStart(1, "a", 1);
        ɵngcc0.ɵɵlistener("click", function DropdownComponent_Template_a_click_1_listener() { return ctx.toggle(); });
        ɵngcc0.ɵɵtext(2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵtemplate(4, DropdownComponent_ng_container_4_Template, 4, 3, "ng-container", 3);
        ɵngcc0.ɵɵprojection(5);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(5, _c2, ctx.isShowDropdown));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵattribute("aria-expanded", ctx.isShowDropdown);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵtextInterpolate1(" ", ctx.buttonText, " ");
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(7, _c2, ctx.isShowDropdown));
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.items);
    } }, directives: [ɵngcc1.NgClass, ɵngcc1.NgForOf, ɵngcc1.NgIf], styles: [""] });
DropdownComponent.ctorParameters = () => [];
DropdownComponent.propDecorators = {
    items: [{ type: Input }],
    buttonText: [{ type: Input }],
    isOpen: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DropdownComponent, [{
        type: Component,
        args: [{
                selector: 'ui-dropdown',
                template: `<div class="dropdown show" [ngClass]="{'show': isShowDropdown}">
  <a class="btn btn-secondary dropdown-toggle" (click)="toggle()" role="button" id="dropdownMenuLink" data-toggle="dropdown"
     aria-haspopup="true" attr.aria-expanded="{{isShowDropdown}}">
    {{buttonText}}
  </a>
  <div class="dropdown-menu dropdown-menu-right" [ngClass]="{'show': isShowDropdown}" aria-labelledby="dropdownMenuLink">
    <ng-container *ngFor="let item of items">
      <a class="dropdown-item" href="{{item.url}}" >{{item.text}}</a>
      <div class="dropdown-divider" *ngIf="item.isDivider"></div>
    </ng-container>
    <ng-content></ng-content>
  </div>
</div>
`,
                styles: [``]
            }]
    }], function () { return []; }, { items: [{
            type: Input
        }], buttonText: [{
            type: Input
        }], isOpen: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DropdownModule {
}
DropdownModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DropdownModule });
DropdownModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DropdownModule_Factory(t) { return new (t || DropdownModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DropdownModule, { declarations: function () { return [DropdownComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [DropdownComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DropdownModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [DropdownComponent],
                exports: [DropdownComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class DatepickerModule {
}
DatepickerModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: DatepickerModule });
DatepickerModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function DatepickerModule_Factory(t) { return new (t || DatepickerModule)(); }, imports: [[
            CommonModule,
            BrowserModule,
            FormsModule,
            DropdownModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(DatepickerModule, { declarations: function () { return [DatepickerComponent]; }, imports: function () { return [CommonModule,
        BrowserModule,
        FormsModule,
        DropdownModule]; }, exports: function () { return [DatepickerComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(DatepickerModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    BrowserModule,
                    FormsModule,
                    DropdownModule
                ],
                declarations: [DatepickerComponent],
                exports: [DatepickerComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class ModalComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
    /**
     * @return {?}
     */
    toggle() {
        this.showModel = !this.showModel;
    }
}
ModalComponent.ɵfac = function ModalComponent_Factory(t) { return new (t || ModalComponent)(); };
ModalComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: ModalComponent, selectors: [["ui-modal"]], ngContentSelectors: _c0, decls: 19, vars: 4, consts: [["type", "button", "data-toggle", "modal", "data-target", "#exampleModal", 1, "btn", "btn-primary", 3, "click"], ["id", "exampleModal", "tabindex", "-1", "role", "dialog", "aria-labelledby", "exampleModalLabel", "aria-hidden", "true", 1, "modal", "fade", 3, "ngClass"], ["role", "document", 1, "modal-dialog"], [1, "modal-content"], [1, "modal-header"], ["id", "exampleModalLabel", 1, "modal-title"], ["type", "button", "data-dismiss", "modal", "aria-label", "Close", 1, "close", 3, "click"], ["aria-hidden", "true"], [1, "modal-body"], [1, "modal-footer"], ["type", "button", "data-dismiss", "modal", 1, "btn", "btn-secondary", 3, "click"], ["type", "button", 1, "btn", "btn-primary", 3, "click"], ["class", "modal-backdrop fade show", 4, "ngIf"], [1, "modal-backdrop", "fade", "show"]], template: function ModalComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "button", 0);
        ɵngcc0.ɵɵlistener("click", function ModalComponent_Template_button_click_0_listener() { return ctx.toggle(); });
        ɵngcc0.ɵɵtext(1, " Launch demo modal\n");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(2, "div", 1);
        ɵngcc0.ɵɵelementStart(3, "div", 2);
        ɵngcc0.ɵɵelementStart(4, "div", 3);
        ɵngcc0.ɵɵelementStart(5, "div", 4);
        ɵngcc0.ɵɵelementStart(6, "h5", 5);
        ɵngcc0.ɵɵtext(7, "Modal title");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(8, "button", 6);
        ɵngcc0.ɵɵlistener("click", function ModalComponent_Template_button_click_8_listener() { return ctx.toggle(); });
        ɵngcc0.ɵɵelementStart(9, "span", 7);
        ɵngcc0.ɵɵtext(10, "\u00D7");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(11, "div", 8);
        ɵngcc0.ɵɵprojection(12);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(13, "div", 9);
        ɵngcc0.ɵɵelementStart(14, "button", 10);
        ɵngcc0.ɵɵlistener("click", function ModalComponent_Template_button_click_14_listener() { return ctx.toggle(); });
        ɵngcc0.ɵɵtext(15, "Close");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(16, "button", 11);
        ɵngcc0.ɵɵlistener("click", function ModalComponent_Template_button_click_16_listener() { return ctx.toggle(); });
        ɵngcc0.ɵɵtext(17, "Save changes");
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵtemplate(18, ModalComponent_div_18_Template, 1, 0, "div", 12);
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(2);
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(2, _c2, ctx.showModel));
        ɵngcc0.ɵɵadvance(16);
        ɵngcc0.ɵɵproperty("ngIf", ctx.showModel);
    } }, directives: [ɵngcc1.NgClass, ɵngcc1.NgIf], styles: [".show[_ngcontent-%COMP%]{display:block;padding-right:16px}"] });
ModalComponent.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ModalComponent, [{
        type: Component,
        args: [{
                selector: 'ui-modal',
                template: `<!-- Button trigger modal -->
<button type="button" class="btn btn-primary" data-toggle="modal" data-target="#exampleModal" (click)="toggle()">
  Launch demo modal
</button>

<!-- Modal -->
<div class="modal fade" id="exampleModal" tabindex="-1" role="dialog" aria-labelledby="exampleModalLabel" aria-hidden="true" [ngClass]="{'show':showModel}">
  <div class="modal-dialog" role="document">
    <div class="modal-content">
      <div class="modal-header">
        <h5 class="modal-title" id="exampleModalLabel">Modal title</h5>
        <button type="button" class="close" data-dismiss="modal" aria-label="Close" (click)="toggle()">
          <span aria-hidden="true">&times;</span>
        </button>
      </div>
      <div class="modal-body">
        <ng-content></ng-content>
      </div>
      <div class="modal-footer">
        <button type="button" class="btn btn-secondary" data-dismiss="modal" (click)="toggle()">Close</button>
        <button type="button" class="btn btn-primary" (click)="toggle()">Save changes</button>
      </div>
    </div>
  </div>
</div>
<div class="modal-backdrop fade show" *ngIf="showModel"></div>
`,
                styles: [`.show{display:block;padding-right:16px}`]
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class ModalModule {
}
ModalModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: ModalModule });
ModalModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function ModalModule_Factory(t) { return new (t || ModalModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(ModalModule, { declarations: function () { return [ModalComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [ModalComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(ModalModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [ModalComponent],
                exports: [ModalComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class TabPanelComponent {
    constructor() {
    }
    /**
     * @return {?}
     */
    ngOnChanges() {
        if (this.isOpen) {
            console.log(this.isOpen);
        }
    }
    /**
     * @return {?}
     */
    ngOnInit() {
        this.index = Math.ceil(1000 * Math.random());
    }
}
TabPanelComponent.ɵfac = function TabPanelComponent_Factory(t) { return new (t || TabPanelComponent)(); };
TabPanelComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TabPanelComponent, selectors: [["ui-tab-panel"]], inputs: { header: "header", isOpen: "isOpen" }, features: [ɵngcc0.ɵɵNgOnChangesFeature], ngContentSelectors: _c0, decls: 2, vars: 4, consts: [["role", "tabpanel", "aria-labelledby", "home-tab", 1, "tab-pane", 3, "ngClass"]], template: function TabPanelComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "div", 0);
        ɵngcc0.ɵɵprojection(1);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵproperty("ngClass", ɵngcc0.ɵɵpureFunction1(2, _c8, ctx.isOpen));
        ɵngcc0.ɵɵattribute("id", "tab" + ctx.index);
    } }, directives: [ɵngcc1.NgClass], styles: [".tab-pane[_ngcontent-%COMP%]{display:none}.tab-pane.active[_ngcontent-%COMP%]{display:block}"] });
TabPanelComponent.ctorParameters = () => [];
TabPanelComponent.propDecorators = {
    header: [{ type: Input }],
    isOpen: [{ type: Input }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TabPanelComponent, [{
        type: Component,
        args: [{
                selector: 'ui-tab-panel',
                template: `<div class="tab-pane" [attr.id]="'tab'+index" [ngClass]="{'active show':isOpen}" role="tabpanel"
     aria-labelledby="home-tab">
  <ng-content></ng-content>
</div>
`,
                styles: [`.tab-pane{display:none}.tab-pane.active{display:block}`]
            }]
    }], function () { return []; }, { header: [{
            type: Input
        }], isOpen: [{
            type: Input
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class TabComponent {
    constructor() {
        this.optionTabAfter = 3;
        this.isShowDropdown = false;
    }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
    /**
     * @return {?}
     */
    toggleDropdown() {
        this.isShowDropdown = !this.isShowDropdown;
    }
    /**
     * @param {?} i
     * @return {?}
     */
    toggle(i) {
        this.tabPanels.forEach(tab => {
            tab.isOpen = (tab.index === i);
        });
    }
    /**
     * @return {?}
     */
    ngAfterContentInit() {
        this.tabs = [];
        this.optionTabs = [];
        /** @type {?} */
        let count = 0;
        this.tabPanels.map(tab => {
            /** @type {?} */
            const tabItem = {
                headerText: tab.header,
                index: tab.index,
                isOpen: !!tab
            };
            count++;
            this.optionTabAfter = (this.optionTabAfter >= this.tabPanels.length) ? this.tabPanels.length : this.optionTabAfter;
            if (count <= this.optionTabAfter) {
                this.tabs.push(tabItem);
            }
            else {
                this.optionTabs.push(tabItem);
            }
        });
        console.log(this.tabs, this.optionTabs);
    }
}
TabComponent.ɵfac = function TabComponent_Factory(t) { return new (t || TabComponent)(); };
TabComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TabComponent, selectors: [["ui-tab"]], contentQueries: function TabComponent_ContentQueries(rf, ctx, dirIndex) { if (rf & 1) {
        ɵngcc0.ɵɵcontentQuery(dirIndex, TabPanelComponent, false);
    } if (rf & 2) {
        var _t;
        ɵngcc0.ɵɵqueryRefresh(_t = ɵngcc0.ɵɵloadQuery()) && (ctx.tabPanels = _t);
    } }, inputs: { optionTabAfter: "optionTabAfter" }, ngContentSelectors: _c0, decls: 5, vars: 2, consts: [["id", "myTab", "role", "tablist", 1, "nav", "nav-tabs"], ["class", "nav-item", 4, "ngFor", "ngForOf"], ["role", "presentation", "class", "dropdown", 3, "ngClass", 4, "ngIf"], ["id", "myTabContent", 1, "tab-content"], [1, "nav-item"], ["data-toggle", "tab", "role", "tab", "aria-controls", "home", 1, "nav-link", 3, "ngClass", "click"], ["role", "presentation", 1, "dropdown", 3, "ngClass"], ["id", "myTabDrop", "data-toggle", "dropdown", "aria-controls", "myTabDrop-contents", 1, "dropdown-toggle", 3, "click"], [1, "caret"], ["aria-labelledby", "myTabDrop", "id", "myTabDrop1-contents", 1, "dropdown-menu", 3, "ngClass"], [4, "ngFor", "ngForOf"], ["data-toggle", "tab", "role", "tab", "aria-controls", "home", 3, "ngClass", "click"]], template: function TabComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵprojectionDef();
        ɵngcc0.ɵɵelementStart(0, "ul", 0);
        ɵngcc0.ɵɵtemplate(1, TabComponent_li_1_Template, 3, 6, "li", 1);
        ɵngcc0.ɵɵtemplate(2, TabComponent_li_2_Template, 6, 8, "li", 2);
        ɵngcc0.ɵɵelementEnd();
        ɵngcc0.ɵɵelementStart(3, "div", 3);
        ɵngcc0.ɵɵprojection(4);
        ɵngcc0.ɵɵelementEnd();
    } if (rf & 2) {
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngForOf", ctx.tabs);
        ɵngcc0.ɵɵadvance(1);
        ɵngcc0.ɵɵproperty("ngIf", ctx.optionTabs.length);
    } }, directives: [ɵngcc1.NgForOf, ɵngcc1.NgIf, ɵngcc1.NgClass], styles: [""] });
TabComponent.ctorParameters = () => [];
TabComponent.propDecorators = {
    optionTabAfter: [{ type: Input }],
    tabPanels: [{ type: ContentChildren, args: [TabPanelComponent,] }]
};
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TabComponent, [{
        type: Component,
        args: [{
                selector: 'ui-tab',
                template: `<ul class="nav nav-tabs" id="myTab" role="tablist">
  <li class="nav-item" *ngFor="let item of tabs;index as i">
    <a class="nav-link "
       [attr.id]="'tab'+item.index"
       data-toggle="tab"
       (click)="toggle(item.index)"
       role="tab" aria-controls="home"
       [attr.aria-selected]="item.isOpen"
       [ngClass]="{'active':item.isOpen}">
      {{item.headerText}}
    </a>
  </li>
  <li role="presentation" class="dropdown" *ngIf="optionTabs.length" [ngClass]="{'show': isShowDropdown}">
    <a class="dropdown-toggle" id="myTabDrop" (click)="toggleDropdown()" data-toggle="dropdown" aria-controls="myTabDrop-contents" attr.aria-expanded="{{isShowDropdown}}">more
      <span class="caret"></span>
    </a>
    <ul class="dropdown-menu" aria-labelledby="myTabDrop" id="myTabDrop1-contents" [ngClass]="{'show': isShowDropdown}">
      <li *ngFor="let item of optionTabs">
        <a [attr.id]="'tab'+item.index"
           data-toggle="tab"
           (click)="toggle(item.index)"
           role="tab" aria-controls="home"
           [attr.aria-selected]="item.isOpen"
           [ngClass]="{'active':item.isOpen}">
          {{item.headerText}}</a>
      </li>
    </ul>
  </li>
</ul>
<div class="tab-content" id="myTabContent">
  <ng-content></ng-content>
</div>
`,
                styles: [``]
            }]
    }], function () { return []; }, { optionTabAfter: [{
            type: Input
        }], tabPanels: [{
            type: ContentChildren,
            args: [TabPanelComponent]
        }] }); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class TabModule {
}
TabModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TabModule });
TabModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TabModule_Factory(t) { return new (t || TabModule)(); }, imports: [[
            CommonModule,
            DropdownModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TabModule, { declarations: function () { return [TabComponent,
        TabPanelComponent]; }, imports: function () { return [CommonModule,
        DropdownModule]; }, exports: function () { return [TabComponent,
        TabPanelComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TabModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule,
                    DropdownModule
                ],
                declarations: [TabComponent, TabPanelComponent],
                exports: [TabComponent, TabPanelComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class TooltipComponent {
    constructor() { }
    /**
     * @return {?}
     */
    ngOnInit() {
    }
}
TooltipComponent.ɵfac = function TooltipComponent_Factory(t) { return new (t || TooltipComponent)(); };
TooltipComponent.ɵcmp = ɵngcc0.ɵɵdefineComponent({ type: TooltipComponent, selectors: [["ui-tooltip"]], decls: 2, vars: 0, template: function TooltipComponent_Template(rf, ctx) { if (rf & 1) {
        ɵngcc0.ɵɵelementStart(0, "p");
        ɵngcc0.ɵɵtext(1, " tooltip works!\n");
        ɵngcc0.ɵɵelementEnd();
    } }, styles: [""] });
TooltipComponent.ctorParameters = () => [];
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TooltipComponent, [{
        type: Component,
        args: [{
                selector: 'ui-tooltip',
                template: `<p>
  tooltip works!
</p>
`,
                styles: [``]
            }]
    }], function () { return []; }, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */
class TooltipModule {
}
TooltipModule.ɵmod = ɵngcc0.ɵɵdefineNgModule({ type: TooltipModule });
TooltipModule.ɵinj = ɵngcc0.ɵɵdefineInjector({ factory: function TooltipModule_Factory(t) { return new (t || TooltipModule)(); }, imports: [[
            CommonModule
        ]] });
(function () { (typeof ngJitMode === "undefined" || ngJitMode) && ɵngcc0.ɵɵsetNgModuleScope(TooltipModule, { declarations: function () { return [TooltipComponent]; }, imports: function () { return [CommonModule]; }, exports: function () { return [TooltipComponent]; } }); })();
/*@__PURE__*/ (function () { ɵngcc0.ɵsetClassMetadata(TooltipModule, [{
        type: NgModule,
        args: [{
                imports: [
                    CommonModule
                ],
                declarations: [TooltipComponent],
                exports: [TooltipComponent]
            }]
    }], null, null); })();

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

/**
 * @fileoverview added by tsickle
 * @suppress {checkTypes,extraRequire,missingReturn,unusedPrivateMembers,uselessCode} checked by tsc
 */

export { NgUikitModule, AccordionModule, AutocompleteModule, CardModule, CarouselModule, ChartModule, DatatableModule, DatepickerModule, DropdownModule, ModalModule, TabModule, TooltipModule, AccordionPanelHeaderComponent as ɵd, AccordionPanelComponent as ɵc, AccordionComponent as ɵb, AutocompleteComponent as ɵe, CardComponent as ɵf, CarouselComponent as ɵg, ChartComponent as ɵh, DatatableComponent as ɵj, PaginationComponent as ɵk, DatatableService as ɵi, PagingService as ɵl, DatepickerComponent as ɵn, DatepickerService as ɵo, DropdownComponent as ɵm, ModalComponent as ɵp, NgUikitComponent as ɵa, TabPanelComponent as ɵr, TabComponent as ɵq, TooltipComponent as ɵs };
